
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Bayesian Normal Mean Model &#8212; Statistical backgrounds for the Summer Program in Statistical Functional Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'Bayesian_normal_mean_model';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bayesian Multivariate Normal Mean Model" href="Bayesian_multivariate_normal_mean_model.html" />
    <link rel="prev" title="Bayes Factor" href="Bayes_factor.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="README.html">
  
  
  
  
  
  
    <p class="title logo__title">Statistical backgrounds for the Summer Program in Statistical Functional Genomics</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Notations and References</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Notations.html">Notation</a></li>

</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Popular Topics and Related Notebooks</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="popular_topics.html">Popular topics</a></li>

</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Basic Concepts</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html">Genotype Coding</a></li>





<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html">Minor Allele Frequency</a></li>




<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html">Hardy-Weinberg Equilibrium</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Correlation Between Variants</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html">Linkage Disequilibrium</a></li>





<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html">Linkage Disequilibrium Score</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Correlation Between Individuals</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html">Genetic Relationship Matrix</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Fixed Effects Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html">Ordinary Least Squares</a></li>





<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html">Odds Ratio</a></li>





<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html">Marginal and Joint Effects</a></li>




<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html">Summary Statistics</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Random Effects Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="random_effect.html">Random Effect</a></li>





<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html">Proportion of Variance Explained and Heritability</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Linear Mixed Model</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html">Linear Mixed Model</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Covariates</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="confounder.html">Confounder</a></li>




<li class="toctree-l1"><a class="reference internal" href="collider.html">Collider</a></li>




<li class="toctree-l1"><a class="reference internal" href="mediator.html">Mediator</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Meta-analysis</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="meta_analysis_fixed_effect.html">Fixed Effects Meta-Analysis</a></li>





<li class="toctree-l1"><a class="reference internal" href="meta_analysis_random_effect.html">Random Effects Meta-Analysis</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Likelihood</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="likelihood.html">Likelihood</a></li>






<li class="toctree-l1"><a class="reference internal" href="maximum_likelihood_estimation.html">Maximum Likelihood Estimation</a></li>





<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html">Likelihood Ratio</a></li>







<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html">Likelihood Ratio Test</a></li>







</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Bayesian Versus Frequentist</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayes_rule.html">Bayes Rule</a></li>





<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html">Bayes Factor</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Bayes Models</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Bayesian Normal Mean Model</a></li>





<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html">Bayesian Multivariate Normal Mean Model</a></li>





<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html">Bayesian Mixture Model</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Model Comparison</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html">Bayesian Model Comparison</a></li>





</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/Bayesian_normal_mean_model.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Bayesian Normal Mean Model</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Bayesian Normal Mean Model</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#graphical-summary">Graphical Summary</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#key-formula">Key Formula</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#technical-details">Technical Details</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#model-assumptions">Model Assumptions</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#prior-distribution">Prior Distribution</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#likelihood">Likelihood</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#posterior-distribution">Posterior Distribution</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#characteristics">Characteristics</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#example">Example</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generate-observed-data">Generate Observed Data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bayesian-inference">Bayesian Inference</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Prior Distribution</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Likelihood</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#posterior">Posterior</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#optional-compare-with-mle">(optional) Compare with MLE</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#supplementary">Supplementary</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">Graphical Summary</a></li>
</ul>
</li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="bayesian-normal-mean-model">
<h1>Bayesian Normal Mean Model<a class="headerlink" href="#bayesian-normal-mean-model" title="Link to this heading">#</a></h1>
<p>The Bayesian normal mean model combines <strong>our prior belief</strong> about a normal distribution’s mean with <strong>observed data</strong>, resulting in a posterior mean that’s a precision-weighted average of the prior mean and sample mean.</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="graphical-summary">
<h1>Graphical Summary<a class="headerlink" href="#graphical-summary" title="Link to this heading">#</a></h1>
<p><img alt="fig" src="graphical_summary/Slide25.png" /></p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="key-formula">
<h1>Key Formula<a class="headerlink" href="#key-formula" title="Link to this heading">#</a></h1>
<p>Under the Bayesian regression setting, we have
$<span class="math notranslate nohighlight">\(
\mathbf{Y} = \mathbf{X} \beta + \boldsymbol{\varepsilon}, \quad \boldsymbol{\varepsilon} \sim \mathcal{N}(0, \sigma^2 \mathbf{I})
\)</span><span class="math notranslate nohighlight">\(
where we place a **prior distribution** on the regression coefficients: \)</span>\beta \sim \mathcal{N}(\beta_0, \Sigma_0)$.</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\( \mathbf{Y} \in \mathbb{R}^{N \times 1} \)</span>: phenotype vector</p></li>
<li><p><span class="math notranslate nohighlight">\( \mathbf{X} \in \mathbb{R}^{N \times 1} \)</span>: genotype vector (e.g., 0, 1, 2 for additive coding)</p></li>
<li><p><span class="math notranslate nohighlight">\( \beta \in \mathbb{R} \)</span>: a scalar, genetic effect size (regression coefficient)</p></li>
<li><p><span class="math notranslate nohighlight">\( \boldsymbol{\varepsilon} \in \mathbb{R}^{N \times 1} \)</span>: residual error, assumed i.i.d. Gaussian</p></li>
</ul>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="technical-details">
<h1>Technical Details<a class="headerlink" href="#technical-details" title="Link to this heading">#</a></h1>
<p>Note that we switch from <span class="math notranslate nohighlight">\(\beta\)</span> to <span class="math notranslate nohighlight">\(\beta\)</span> as we now consider <span class="math notranslate nohighlight">\(\beta\)</span> a random variable to estimate, but conceptually they are the same thing – effect of <span class="math notranslate nohighlight">\(X\)</span> on <span class="math notranslate nohighlight">\(Y\)</span>.</p>
<section id="model-assumptions">
<h2>Model Assumptions<a class="headerlink" href="#model-assumptions" title="Link to this heading">#</a></h2>
<p>In the Bayesian normal mean model, we assume the variable <span class="math notranslate nohighlight">\(Y\)</span> follows uni-variate normal model <span class="math notranslate nohighlight">\(\mathcal{N}(X\beta, \sigma^2)\)</span></p>
<div class="math notranslate nohighlight">
\[
Y \sim \mathcal{N}(X\beta, \sigma^2)
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\( \beta \)</span> is the unknown mean parameter.</p></li>
<li><p><span class="math notranslate nohighlight">\( \sigma^2 \)</span> is the known variance.</p></li>
<li><p><span class="math notranslate nohighlight">\(X\)</span> is the genotype value (fixed) for each individual</p></li>
</ul>
<p>Since <span class="math notranslate nohighlight">\(X\)</span> is fixed, then the question becomes what can we learn about the unknown parameter <span class="math notranslate nohighlight">\(\beta\)</span>.</p>
</section>
<section id="prior-distribution">
<h2>Prior Distribution<a class="headerlink" href="#prior-distribution" title="Link to this heading">#</a></h2>
<p>A common choice for the prior on <span class="math notranslate nohighlight">\( \beta \)</span> is a normal distribution:</p>
<div class="math notranslate nohighlight">
\[
\beta \sim \mathcal{N}(\beta_0, \sigma_0^2)
\]</div>
<p>where <span class="math notranslate nohighlight">\( \beta_0 \)</span> is the prior mean and <span class="math notranslate nohighlight">\( \sigma_0^2 \)</span> is the prior variance.</p>
<p>The precision is defined as <span class="math notranslate nohighlight">\(\tau_0 = \frac{1}{\sigma_0^2}\)</span>. In other words, the prior of <span class="math notranslate nohighlight">\(\beta\)</span> can be rewritten as:</p>
<div class="math notranslate nohighlight">
\[
\beta \sim \mathcal{N}(\beta_0, \frac{1}{\tau_0})
\]</div>
<p>Therefore, the probability density function (PDF) is</p>
<div class="math notranslate nohighlight">
\[\begin{split}
\begin{align}
p(\beta)&amp;\propto \exp\left(-\frac{1}{2}\tau_0\beta^2 + \tau_0\beta_0 \beta\right) \\
       &amp;\propto \exp\left(-\frac{1}{2}\tau_0 (\beta - \beta_0)^2\right)
\end{align}
\end{split}\]</div>
<p>(Here, the 0 subscript is used to indicate that <span class="math notranslate nohighlight">\(\beta_0\)</span> and <span class="math notranslate nohighlight">\(\sigma_0\)</span> are parameters in the prior.)</p>
</section>
<section id="likelihood">
<h2>Likelihood<a class="headerlink" href="#likelihood" title="Link to this heading">#</a></h2>
<p>The likelihood of the data given <span class="math notranslate nohighlight">\(\beta\)</span> and <span class="math notranslate nohighlight">\(\sigma^2\)</span> is:</p>
<div class="math notranslate nohighlight">
\[
Y\mid\beta, \sigma^2 \sim \mathcal{N}(\beta, \sigma^2)
\]</div>
<p>The probability density function (PDF) is:</p>
<div class="math notranslate nohighlight">
\[\begin{split}
\begin{align}
p(Y\mid\beta) &amp;\propto \exp\left(-\frac{1}{2} \tau (Y^2 - 2\beta Y)\right)\\
&amp;\propto \exp\left(-\frac{1}{2} \tau (Y - \beta)^2\right)
\end{align}
\end{split}\]</div>
<p>where <span class="math notranslate nohighlight">\(\tau = \frac{1}{\sigma^2}\)</span> is the precision.</p>
</section>
<section id="posterior-distribution">
<h2>Posterior Distribution<a class="headerlink" href="#posterior-distribution" title="Link to this heading">#</a></h2>
<p>Using Bayes’ theorem, the posterior distribution of <span class="math notranslate nohighlight">\( \beta \)</span> is given by:</p>
<div class="math notranslate nohighlight">
\[
p(\beta|Y) \propto p(Y|\beta) p(\beta)
\]</div>
<p>Expanding the probability density functions:</p>
<div class="math notranslate nohighlight">
\[\begin{split}
\begin{align}
p(\beta \mid Y) &amp;\propto \exp\left(-\frac{1}{2} \tau (Y - \beta)^2\right) \exp\left(-\frac{1}{2} \tau_0 (\beta - \beta_0)^2\right) \\
&amp;\propto \exp\left(-\frac{1}{2} \tau (Y^2 - 2Y\beta + \beta^2) - \frac{1}{2} \tau_0 (\beta^2 - 2\beta \beta_0 + \beta_0^2)\right) \\
&amp;\propto \exp\left(-\frac{1}{2} (\tau + \tau_0) \beta^2 + (\tau Y + \tau_0 \beta_0) \beta \right)
\end{align}
\end{split}\]</div>
<p>Recognizing the quadratic form, the posterior distribution follows a normal distribution:</p>
<div class="math notranslate nohighlight">
\[
\beta \mid Y \sim \mathcal{N}\left( \beta_1, \sigma_1^2 \right)
\]</div>
<p>where</p>
<div class="math notranslate nohighlight">
\[
\beta_1 = \frac{\tau Y + \tau_0 \beta_0}{\tau + \tau_0}, \quad \sigma_1^2 = \frac{1}{\tau + \tau_0}
\]</div>
</section>
<section id="characteristics">
<h2>Characteristics<a class="headerlink" href="#characteristics" title="Link to this heading">#</a></h2>
<ul>
<li><p><strong>Bayes’ theorem</strong> combines these components: <span class="math notranslate nohighlight">\(P(\beta|D) \propto P(D|\beta) P(\beta)\)</span>.</p></li>
<li><p>The <strong>prior</strong> represents our belief about the parameter before seeing any data.</p></li>
<li><p>The <strong>likelihood</strong> represents how likely we are to observe the data given the parameter.</p></li>
<li><p>The <strong>posterior</strong> is the updated belief about the parameter after seeing the data.</p>
<ul class="simple">
<li><p>the <strong>Posterior precision</strong> (<span class="math notranslate nohighlight">\(\tau_1\)</span>) is the sum of the Data precision (<span class="math notranslate nohighlight">\(\tau\)</span>) and the Prior precision (<span class="math notranslate nohighlight">\(\tau_0\)</span>). This makes sense: the more precise your data, and the more precise your prior information, the more precise your posterior information. Also, this means that the data always improves your posterior precision compared with the prior: noisy data (small <span class="math notranslate nohighlight">\(\tau\)</span>) improves it only a little, whereas precise data improves it a lot.</p></li>
<li><p>If we rewrite the <strong>posterior mean</strong> as:</p></li>
</ul>
<div class="math notranslate nohighlight">
\[
    \beta_1=w\sum_i^n x_i+(1−w)\beta_0
    \]</div>
<p>where <span class="math notranslate nohighlight">\(w=\tau(\tau+\tau_0)\)</span>. Thus <span class="math notranslate nohighlight">\(\beta_1\)</span> is a weighted average of the sum of the data <span class="math notranslate nohighlight">\(x_i\)</span> and the prior mean <span class="math notranslate nohighlight">\(\beta_0\)</span>. And the weights depend on the relative precision of the data and the prior. If the data are precise compared with the prior (<span class="math notranslate nohighlight">\(\tau&gt;&gt;\tau_0\)</span>) then the weight <span class="math notranslate nohighlight">\(w\)</span> will be close to 1 and the posterior mean will be close to the data.</p>
</li>
</ul>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="example">
<h1>Example<a class="headerlink" href="#example" title="Link to this heading">#</a></h1>
<p>This R example demonstrates a Bayesian normal mean model for updating beliefs about a population parameter using observed data. The analysis starts with a prior belief that the true mean (<span class="math notranslate nohighlight">\(\beta\)</span>) follows a normal distribution with mean 0.1 and variance 1. After observing height measurements from five individuals, the code applies Bayesian updating rules to compute the posterior distribution, which combines the prior information with the likelihood of the observed data. The example illustrates how the posterior mean shifts toward the sample average while the posterior variance decreases (indicating increased certainty), and visualizes both the prior and posterior distributions along with the observed data points to show how Bayesian inference updates our beliefs in light of new evidence.</p>
<ul class="simple">
<li><p>Requirement:</p>
<ul>
<li><p><a class="reference external" href="https://gaow.github.io/statgen-prerequisites/Bayes_rule.html">Bayes rule</a></p></li>
<li><p><a class="reference external" href="https://gaow.github.io/statgen-prerequisites/ordinary_least_squarehtml">OLS</a></p></li>
</ul>
</li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Clear the environment</span>
<span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ls</span><span class="p">())</span>
</pre></div>
</div>
</div>
</div>
<section id="generate-observed-data">
<h2>Generate Observed Data<a class="headerlink" href="#generate-observed-data" title="Link to this heading">#</a></h2>
<p>First we define the true parameters, which are unknown in practice, but here we are simulating the data. Then we generate the observed data.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># True genetic effect of one variant on height</span>
<span class="n">beta_true</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">4.5</span><span class="w">      </span><span class="c1"># cm per allele (this is what we want to estimate)</span>
<span class="n">sigma_epsilon</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">8</span><span class="w">    </span><span class="c1"># residual standard deviation (cm)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">n_individuals</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">10</span><span class="w">           </span><span class="c1"># number of individuals</span>
<span class="n">height_baseline</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">170</span><span class="w">       </span><span class="c1"># population mean height (cm)</span>

<span class="c1"># Generate genotypes (0, 1, or 2 copies of the variant)</span>
<span class="c1"># Assuming MAF = 0.3 (Minor Allele Frequency)</span>
<span class="n">maf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0.3</span>
<span class="n">p_genotypes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">((</span><span class="m">1</span><span class="o">-</span><span class="n">maf</span><span class="p">)</span><span class="o">^</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="o">*</span><span class="n">maf</span><span class="o">*</span><span class="p">(</span><span class="m">1</span><span class="o">-</span><span class="n">maf</span><span class="p">),</span><span class="w"> </span><span class="n">maf</span><span class="o">^</span><span class="m">2</span><span class="p">)</span><span class="w">  </span><span class="c1"># P(0), P(1), P(2)</span>

<span class="n">genotypes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sample</span><span class="p">(</span><span class="m">0</span><span class="o">:</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n_individuals</span><span class="p">,</span><span class="w"> </span><span class="n">replace</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">prob</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p_genotypes</span><span class="p">)</span>

<span class="c1"># Calculate genetic contributions and add environmental noise</span>
<span class="n">genetic_effects</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">genotypes</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">beta_true</span>
<span class="n">environmental_effects</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rnorm</span><span class="p">(</span><span class="n">n_individuals</span><span class="p">,</span><span class="w"> </span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sigma_epsilon</span><span class="p">)</span>
<span class="n">observed_heights</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">height_baseline</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">genetic_effects</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">environmental_effects</span>

<span class="c1"># Create dataset</span>
<span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">  </span><span class="n">individual</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">n_individuals</span><span class="p">,</span>
<span class="w">  </span><span class="n">genotype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genotypes</span><span class="p">,</span>
<span class="w">  </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">observed_heights</span><span class="p">,</span>
<span class="w">  </span><span class="n">genetic_effect</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genetic_effects</span><span class="p">,</span>
<span class="w">  </span><span class="n">environmental_effect</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">environmental_effects</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="bayesian-inference">
<h2>Bayesian Inference<a class="headerlink" href="#bayesian-inference" title="Link to this heading">#</a></h2>
<section id="id1">
<h3>Prior Distribution<a class="headerlink" href="#id1" title="Link to this heading">#</a></h3>
<p>We assume a normal prior for <span class="math notranslate nohighlight">\(\beta\)</span>:</p>
<div class="math notranslate nohighlight">
\[
\beta \sim \mathcal{N}(\beta_0, \sigma_0^2)
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(\beta_0 = 0\)</span> (prior mean),</p></li>
<li><p><span class="math notranslate nohighlight">\(\sigma_0^2 = 25\)</span> (prior variance), leading to a precision <span class="math notranslate nohighlight">\(\tau_0 = 1/\sigma_0^2\)</span>.</p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Prior belief about genetic effect (weakly informative)</span>
<span class="n">beta_prior_mean</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">         </span><span class="c1"># center at zero (no prior knowledge)</span>
<span class="n">beta_prior_var</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">25</span><span class="w">         </span><span class="c1"># prior variance (SD = 5 cm per allele)</span>
<span class="n">beta_prior_precision</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">beta_prior_var</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="id2">
<h3>Likelihood<a class="headerlink" href="#id2" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Likelihood for each individual i:\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;height_i | beta ~ N(&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">height_baseline</span><span class="p">,</span><span class="w"> </span><span class="s">&quot; + beta * genotype_i,&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sigma_epsilon</span><span class="o">^</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;)\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Known residual variance:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sigma_epsilon</span><span class="o">^</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cm^2\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Residual precision:&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">/</span><span class="n">sigma_epsilon</span><span class="o">^</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n\n&quot;</span><span class="p">)</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;The likelihood is a function of beta (unknown parameter):\n&quot;</span><span class="p">)</span>
<span class="kr">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">min</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">n_individuals</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Individual&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;: height =&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">height</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span>
<span class="w">      </span><span class="s">&quot;cm, genotype =&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">genotype</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;  Expected height = &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">height_baseline</span><span class="p">,</span><span class="w"> </span><span class="s">&quot; + beta *&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">genotype</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span>
<span class="w">      </span><span class="s">&quot;=&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">height_baseline</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;+ beta *&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">genotype</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;  Likelihood contribution is proportional to exp(-0.5 * (&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">height</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span>
<span class="w">      </span><span class="s">&quot;- (&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">height_baseline</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;+ beta *&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">genotype</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&quot;))^2 /&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sigma_epsilon</span><span class="o">^</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;)\n&quot;</span><span class="p">)</span>
<span class="p">}</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;...\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;The likelihood function tells us how plausible different values of beta are,\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;given the observed data. We don&#39;t know the true beta - that&#39;s what we&#39;re estimating!\n\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Likelihood for each individual i:
height_i | beta ~ N( 170  + beta * genotype_i, 64 )
Known residual variance: 64 cm^2
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Residual precision: 0.015625 

The likelihood is a function of beta (unknown parameter):
Individual 1 : height = 172.6 cm, genotype = 2 
  Expected height =  170  + beta * 2 = 170 + beta * 2 
  Likelihood contribution is proportional to exp(-0.5 * ( 172.6 - ( 170 + beta * 2 ))^2 / 64 )
Individual 2 : height = 174 cm, genotype = 0 
  Expected height =  170  + beta * 0 = 170 + beta * 0 
  Likelihood contribution is proportional to exp(-0.5 * ( 174 - ( 170 + beta * 0 ))^2 / 64 )
Individual 3 : height = 172 cm, genotype = 0 
  Expected height =  170  + beta * 0 = 170 + beta * 0 
  Likelihood contribution is proportional to exp(-0.5 * ( 172 - ( 170 + beta * 0 ))^2 / 64 )
...
The likelihood function tells us how plausible different values of beta are,
given the observed data. We don&#39;t know the true beta - that&#39;s what we&#39;re estimating!
</pre></div>
</div>
</div>
</div>
<p>A <strong>sufficient statistic</strong> T(data) contains all the information about a parameter that is available in the dataset. Remarkably, for our normal model, ALL information about β from n individuals can be compressed into just <strong>two numbers</strong>:</p>
<ul class="simple">
<li><p><strong><span class="math notranslate nohighlight">\(T_1\)</span></strong> = Σ(genotype_i²)</p></li>
<li><p><strong><span class="math notranslate nohighlight">\(T_2\)</span></strong> = Σ(genotype_i · (height_i - baseline))</p></li>
</ul>
<p>Due to the <strong>exponential family</strong> property of the normal distribution, the likelihood can be rewritten as:</p>
<div class="math notranslate nohighlight">
\[
L(\beta|\text{data}) = L(\beta|T_1, T_2, N)
\]</div>
<p>This means we can <strong>throw away all the raw data</strong> and work only with <span class="math notranslate nohighlight">\(T_1\)</span>, <span class="math notranslate nohighlight">\(T_2\)</span>, and <span class="math notranslate nohighlight">\(N\)</span>, without losing any information about <span class="math notranslate nohighlight">\(\beta\)</span>!</p>
<p>Sufficient statistics are particularly powerful when combined with <strong>conjugate priors</strong>. For the normal-normal model:</p>
<ul class="simple">
<li><p>Prior: <span class="math notranslate nohighlight">\(\beta \sim N(\beta_\text{prior}, \sigma_\text{prior}^2)\)</span></p></li>
<li><p>Likelihood: data | <span class="math notranslate nohighlight">\(\beta\)</span> ~ Normal with sufficient statistics <span class="math notranslate nohighlight">\(T_1, T_2\)</span></p></li>
<li><p>Posterior: <span class="math notranslate nohighlight">\(\beta | \text{data} \sim N(\beta_\text{posterior}, \sigma_\text{posterior}^2))\)</span> with simple update formulas</p></li>
</ul>
<p>This creates a beautiful mathematical framework where:</p>
<ol class="arabic simple">
<li><p>We can summarize any amount of data with just a few numbers</p></li>
<li><p>Bayesian updating becomes elementary algebra</p></li>
<li><p>The posterior has the same functional form as the prior</p></li>
</ol>
<p>Sufficient statistics transform Bayesian inference from a computationally intensive, data-heavy process into an elegant, efficient calculation. They are the reason why certain Bayesian models can scale to massive datasets while maintaining mathematical rigor and computational tractability.</p>
<p>Without sufficient statistics, modern Bayesian genomics, machine learning, and large-scale data analysis would be practically impossible.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># For conjugate normal-normal model, we need:</span>
<span class="c1"># Sum of (genotype_i^2) and Sum of (genotype_i * (height_i - baseline))</span>

<span class="n">sum_genotype_squared</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">genotypes</span><span class="o">^</span><span class="m">2</span><span class="p">)</span>
<span class="n">sum_genotype_height</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">genotypes</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="n">observed_heights</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">height_baseline</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="posterior">
<h3>Posterior<a class="headerlink" href="#posterior" title="Link to this heading">#</a></h3>
<p>Using Bayesian updating, the posterior distribution is given by:</p>
<div class="math notranslate nohighlight">
\[
\beta | Y \sim \mathcal{N}(\beta_1, \sigma_1^2)
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><strong>Posterior precision</strong>:
<span class="math notranslate nohighlight">\(\tau_1 = \tau_0 + n \tau\)</span></p></li>
<li><p><strong>Posterior mean</strong>:
<span class="math notranslate nohighlight">\(\beta_1 = \frac{\tau_0 \beta_0 + \tau \sum Y}{\tau_1}\)</span></p></li>
<li><p><strong>Posterior standard deviation</strong>:
<span class="math notranslate nohighlight">\(\sigma_1 = \sqrt{\frac{1}{\tau_1}}\)</span></p></li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Posterior precision = prior precision + likelihood precision</span>
<span class="n">likelihood_precision</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sum_genotype_squared</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">sigma_epsilon</span><span class="o">^</span><span class="m">2</span>
<span class="n">posterior_precision</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">beta_prior_precision</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">likelihood_precision</span>

<span class="c1"># Posterior variance</span>
<span class="n">posterior_variance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">posterior_precision</span>

<span class="c1"># Posterior mean (precision-weighted average)</span>
<span class="n">posterior_mean</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">beta_prior_precision</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">beta_prior_mean</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">                  </span><span class="n">likelihood_precision</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="n">sum_genotype_height</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">sum_genotype_squared</span><span class="p">))</span><span class="w"> </span><span class="o">/</span><span class="w"> </span>
<span class="w">                  </span><span class="n">posterior_precision</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Bayesian updating formulas:\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Posterior precision = prior precision + likelihood precision\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;                   =&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">beta_prior_precision</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;+&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">likelihood_precision</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span>
<span class="w">    </span><span class="s">&quot;=&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">posterior_precision</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n\n&quot;</span><span class="p">)</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Posterior variance = 1 / posterior precision =&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">posterior_variance</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n\n&quot;</span><span class="p">)</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Posterior mean = weighted average of prior and data\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Weight on prior:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">beta_prior_precision</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">posterior_precision</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Weight on data:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">likelihood_precision</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">posterior_precision</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n\n&quot;</span><span class="p">)</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;POSTERIOR RESULTS:\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Posterior distribution: beta | data ~ N(&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">posterior_mean</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">    </span><span class="nf">round</span><span class="p">(</span><span class="n">posterior_variance</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;)\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Posterior mean:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">posterior_mean</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;cm per allele\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Posterior standard deviation:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="nf">sqrt</span><span class="p">(</span><span class="n">posterior_variance</span><span class="p">),</span><span class="w"> </span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;cm per allele\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;True value (for comparison):&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">beta_true</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cm per allele\n\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Bayesian updating formulas:
Posterior precision = prior precision + likelihood precision
                   = 0.04 + 0.0938 = 0.1338 

Posterior variance = 1 / posterior precision = 7.4766 

Posterior mean = weighted average of prior and data
Weight on prior: 0.299 
Weight on data: 0.701 

POSTERIOR RESULTS:
Posterior distribution: beta | data ~ N( 2.215 , 7.4766 )
Posterior mean: 2.215 cm per allele
Posterior standard deviation: 2.734 cm per allele
True value (for comparison): 4.5 cm per allele
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="optional-compare-with-mle">
<h2>(optional) Compare with MLE<a class="headerlink" href="#optional-compare-with-mle" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Maximum likelihood estimator (same as OLS for normal model)</span>
<span class="c1"># beta_hat = (X&#39;X)^(-1) X&#39;Y where X is design matrix and Y is outcome vector</span>

<span class="c1"># For our simple model: height_i = baseline + beta * genotype_i + epsilon_i</span>
<span class="c1"># The MLE is: beta_hat = sum(genotype_i * (height_i - baseline)) / sum(genotype_i^2)</span>

<span class="n">beta_mle</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sum_genotype_height</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">sum_genotype_squared</span>
<span class="n">se_mle</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sqrt</span><span class="p">(</span><span class="n">sigma_epsilon</span><span class="o">^</span><span class="m">2</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">sum_genotype_squared</span><span class="p">)</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Maximum Likelihood Estimator (MLE):\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;beta_hat =&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">beta_mle</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;cm per allele\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Standard error:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">se_mle</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;cm per allele\n&quot;</span><span class="p">)</span>

<span class="c1"># 95% confidence interval</span>
<span class="n">ci_lower</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">beta_mle</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">1.96</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">se_mle</span>
<span class="n">ci_upper</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">beta_mle</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1.96</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">se_mle</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;95% Confidence Interval: [&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">ci_lower</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">ci_upper</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;] cm per allele\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Maximum Likelihood Estimator (MLE):
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>beta_hat = 3.16 cm per allele
Standard error: 3.266 cm per allele
95% Confidence Interval: [ -3.24 , 9.56 ] cm per allele
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="supplementary">
<h1>Supplementary<a class="headerlink" href="#supplementary" title="Link to this heading">#</a></h1>
<section id="id3">
<h2>Graphical Summary<a class="headerlink" href="#id3" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>

<span class="c1"># Set parameters</span>
<span class="c1"># Prior: N(mu_0, sigma_0^2)</span>
<span class="n">mu_0</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">      </span><span class="c1"># Prior mean</span>
<span class="n">sigma_0</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="c1"># Prior standard deviation (smaller = more informative prior)</span>

<span class="c1"># Data (observed values)</span>
<span class="n">y_obs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="m">2.1</span><span class="p">,</span><span class="w"> </span><span class="m">1.8</span><span class="p">,</span><span class="w"> </span><span class="m">2.3</span><span class="p">,</span><span class="w"> </span><span class="m">1.9</span><span class="p">)</span><span class="w">  </span><span class="c1"># Sample data</span>
<span class="n">n</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">y_obs</span><span class="p">)</span>
<span class="n">y_bar</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">mean</span><span class="p">(</span><span class="n">y_obs</span><span class="p">)</span><span class="w">  </span><span class="c1"># Sample mean</span>
<span class="n">sigma</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">4</span><span class="w">  </span><span class="c1"># Known likelihood standard deviation (larger = less informative data)</span>

<span class="c1"># Calculate posterior parameters (conjugate normal-normal case)</span>
<span class="c1"># Posterior precision = prior precision + likelihood precision</span>
<span class="n">tau_0</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">sigma_0</span><span class="o">^</span><span class="m">2</span><span class="w">    </span><span class="c1"># Prior precision</span>
<span class="n">tau_likelihood</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">sigma</span><span class="o">^</span><span class="m">2</span><span class="w">  </span><span class="c1"># Likelihood precision</span>
<span class="n">tau_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tau_0</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">tau_likelihood</span><span class="w">  </span><span class="c1"># Posterior precision</span>

<span class="c1"># Posterior parameters</span>
<span class="n">sigma_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sqrt</span><span class="p">(</span><span class="m">1</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">tau_1</span><span class="p">)</span><span class="w">  </span><span class="c1"># Posterior standard deviation</span>
<span class="n">mu_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">tau_0</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">mu_0</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">tau_likelihood</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">y_bar</span><span class="p">)</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">tau_1</span><span class="w">  </span><span class="c1"># Posterior mean</span>

<span class="c1"># Create data for plotting</span>
<span class="n">beta_range</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">seq</span><span class="p">(</span><span class="m">-4</span><span class="p">,</span><span class="w"> </span><span class="m">6</span><span class="p">,</span><span class="w"> </span><span class="n">length.out</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">)</span>

<span class="c1"># Calculate densities</span>
<span class="n">prior_density</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dnorm</span><span class="p">(</span><span class="n">beta_range</span><span class="p">,</span><span class="w"> </span><span class="n">mu_0</span><span class="p">,</span><span class="w"> </span><span class="n">sigma_0</span><span class="p">)</span>
<span class="n">likelihood_density</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dnorm</span><span class="p">(</span><span class="n">beta_range</span><span class="p">,</span><span class="w"> </span><span class="n">y_bar</span><span class="p">,</span><span class="w"> </span><span class="n">sigma</span><span class="o">/</span><span class="nf">sqrt</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w">  </span><span class="c1"># Likelihood as function of beta</span>
<span class="n">posterior_density</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dnorm</span><span class="p">(</span><span class="n">beta_range</span><span class="p">,</span><span class="w"> </span><span class="n">mu_1</span><span class="p">,</span><span class="w"> </span><span class="n">sigma_1</span><span class="p">)</span>

<span class="c1"># Combine into data frame</span>
<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">  </span><span class="n">beta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="n">beta_range</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">),</span>
<span class="w">  </span><span class="n">density</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">prior_density</span><span class="p">,</span><span class="w"> </span><span class="n">likelihood_density</span><span class="p">,</span><span class="w"> </span><span class="n">posterior_density</span><span class="p">),</span>
<span class="w">  </span><span class="n">distribution</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Prior&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Likelihood&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Posterior&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">each</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">beta_range</span><span class="p">))</span>
<span class="p">)</span>

<span class="c1"># Create the plot</span>
<span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">beta</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">density</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">distribution</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">distribution</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_line</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Prior&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tomato&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                               </span><span class="s">&quot;Likelihood&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;#000080&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                               </span><span class="s">&quot;Posterior&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;darkgreen&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_linetype_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Prior&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                  </span><span class="s">&quot;Likelihood&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dotted&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                  </span><span class="s">&quot;Posterior&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;solid&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span>
<span class="w">    </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Density&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">expression</span><span class="p">(</span><span class="n">beta</span><span class="p">),</span>
<span class="w">    </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Distribution&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Distribution&quot;</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">(</span><span class="n">base_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span>
<span class="w">    </span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">(),</span>
<span class="w">    </span><span class="n">axis.title.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">axis.title.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">(),</span>
<span class="w">    </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bottom&quot;</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">guides</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">guide_legend</span><span class="p">(</span><span class="n">override.aes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.5</span><span class="p">)))</span>

<span class="c1"># Add vertical lines for means</span>
<span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">p</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mu_0</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tomato&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dashed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y_bar</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;skyblue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;dotted&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mu_1</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;darkgreen&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;solid&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">)</span>

<span class="c1"># Display plot</span>
<span class="nf">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>

<span class="c1"># Save plot to file</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;./cartoons/Bayesian_normal_mean_model.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">plot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p</span><span class="p">,</span>
<span class="w">       </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">bg</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;transparent&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">300</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="_images/37191fec1664c3ca77432a35913e5a7a455677f1505774a5a3bde54f300417d8.png"><img alt="_images/37191fec1664c3ca77432a35913e5a7a455677f1505774a5a3bde54f300417d8.png" src="_images/37191fec1664c3ca77432a35913e5a7a455677f1505774a5a3bde54f300417d8.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="Bayes_factor.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Bayes Factor</p>
      </div>
    </a>
    <a class="right-next"
       href="Bayesian_multivariate_normal_mean_model.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Bayesian Multivariate Normal Mean Model</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Bayesian Normal Mean Model</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#graphical-summary">Graphical Summary</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#key-formula">Key Formula</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#technical-details">Technical Details</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#model-assumptions">Model Assumptions</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#prior-distribution">Prior Distribution</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#likelihood">Likelihood</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#posterior-distribution">Posterior Distribution</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#characteristics">Characteristics</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#example">Example</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generate-observed-data">Generate Observed Data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bayesian-inference">Bayesian Inference</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Prior Distribution</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Likelihood</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#posterior">Posterior</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#optional-compare-with-mle">(optional) Compare with MLE</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#supplementary">Supplementary</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">Graphical Summary</a></li>
</ul>
</li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Gao Wang
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>