
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Linear Mixed Model &#8212; Statistical backgrounds for the Summer Program in Statistical Functional Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'linear_mixed_model';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Confounder" href="confounder.html" />
    <link rel="prev" title="Proportion of Variance Explained and Heritability" href="proportion_of_variance_explained.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="README.html">
  
  
  
  
  
  
    <p class="title logo__title">Statistical backgrounds for the Summer Program in Statistical Functional Genomics</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Notations and References</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Notations.html">Notation</a></li>

</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Popular Topics and Related Notebooks</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="popular_topics.html">Popular topics</a></li>

</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Basic Concepts</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html">Genotype Coding</a></li>





<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html">Minor Allele Frequency</a></li>




<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html">Hardy-Weinberg Equilibrium</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Correlation Between Variants</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html">Linkage Disequilibrium</a></li>





<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html">Linkage Disequilibrium Score</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Correlation Between Individuals</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html">Genetic Relationship Matrix</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Fixed Effects Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html">Ordinary Least Squares</a></li>





<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html">Odds Ratio</a></li>





<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html">Marginal and Joint Effects</a></li>




<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html">Summary Statistics</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Random Effects Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="random_effect.html">Random Effect</a></li>





<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html">Proportion of Variance Explained and Heritability</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Linear Mixed Model</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Linear Mixed Model</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Covariates</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="confounder.html">Confounder</a></li>




<li class="toctree-l1"><a class="reference internal" href="collider.html">Collider</a></li>




<li class="toctree-l1"><a class="reference internal" href="mediator.html">Mediator</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Meta-analysis</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="meta_analysis_fixed_effect.html">Fixed Effects Meta-Analysis</a></li>





<li class="toctree-l1"><a class="reference internal" href="meta_analysis_random_effect.html">Random Effects Meta-Analysis</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Likelihood</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="likelihood.html">Likelihood</a></li>






<li class="toctree-l1"><a class="reference internal" href="maximum_likelihood_estimation.html">Maximum Likelihood Estimation</a></li>





<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html">Likelihood Ratio</a></li>







<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html">Likelihood Ratio Test</a></li>







</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Bayesian Versus Frequentist</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayes_rule.html">Bayes Rule</a></li>





<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html">Bayes Factor</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Bayes Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html">Bayesian Normal Mean Model</a></li>





<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html">Bayesian Multivariate Normal Mean Model</a></li>





<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html">Bayesian Mixture Model</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Model Comparison</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html">Bayesian Model Comparison</a></li>





</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/linear_mixed_model.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Linear Mixed Model</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Linear Mixed Model</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#graphical-summary">Graphical Summary</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#key-formula">Key Formula</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#technical-details">Technical Details</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#key-challenges-in-statistical-genetics">Key Challenges in Statistical Genetics</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#computational-complexity">Computational Complexity</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#high-dimensional-genomic-data">High-Dimensional Genomic Data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#complex-population-structure">Complex Population Structure</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#why-many-specialized-methods-exist">Why Many Specialized Methods Exist</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#computational-efficiency-methods">Computational Efficiency Methods</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#memory-efficient-approaches">Memory-Efficient Approaches</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#statistical-power-optimization">Statistical Power Optimization</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#example">Example</a></li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="linear-mixed-model">
<h1>Linear Mixed Model<a class="headerlink" href="#linear-mixed-model" title="Link to this heading">#</a></h1>
<p>The random effect model in statistical genetics allows us to <strong>account for correlation between individuals</strong> due to shared genetic background (like family relationships) by partitioning variance into <strong>fixed effects</strong> (specific genetic variants weâ€™re testing) and <strong>random effects</strong> (overall genetic similarity within groups).</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="graphical-summary">
<h1>Graphical Summary<a class="headerlink" href="#graphical-summary" title="Link to this heading">#</a></h1>
<p><img alt="LMM" src="graphical_summary/Slide16.png" /></p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="key-formula">
<h1>Key Formula<a class="headerlink" href="#key-formula" title="Link to this heading">#</a></h1>
<p>In the linear mixed model (<strong>mixed</strong> because it incorporate both the <strong>fixed</strong> and the <strong>random</strong> effects), which accurately represent non-independent data structures,</p>
<div class="math notranslate nohighlight">
\[
\mathbf{Y} = \mathbf{X}\boldsymbol{\beta} + \mathbf{Z}\mathbf{u} + \boldsymbol{\epsilon}
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(\mathbf{Y}\)</span> is the <span class="math notranslate nohighlight">\(N \times 1\)</span> vector of phenotypes</p></li>
<li><p><span class="math notranslate nohighlight">\(\mathbf{X}\)</span> is the <span class="math notranslate nohighlight">\(N \times M\)</span> design matrix for fixed effects (e.g., genotypes, covariates)</p></li>
<li><p><span class="math notranslate nohighlight">\(\boldsymbol{\beta}\)</span> is the <span class="math notranslate nohighlight">\(M \times 1\)</span> vector of fixed effect coefficients (unknown, to be estimated)</p></li>
<li><p><span class="math notranslate nohighlight">\(\mathbf{Z}\)</span> is the <span class="math notranslate nohighlight">\(N \times M\)</span> design matrix for random effects</p></li>
<li><p><span class="math notranslate nohighlight">\(\mathbf{u}\)</span> is the <span class="math notranslate nohighlight">\(M \times 1\)</span> vector of random effects, where <span class="math notranslate nohighlight">\(\mathbf{u} \sim N(0, \sigma^2_u\mathbf{G})\)</span> (unknown)</p></li>
<li><p><span class="math notranslate nohighlight">\(\mathbf{G}\)</span> is the relationship matrix between the <span class="math notranslate nohighlight">\(N\)</span> individuals</p></li>
<li><p><span class="math notranslate nohighlight">\(\boldsymbol{\epsilon}\)</span> is the <span class="math notranslate nohighlight">\(N \times 1\)</span> vector of residual errors, where <span class="math notranslate nohighlight">\(\boldsymbol{\epsilon} \sim N(0, \sigma^2_e\mathbf{I})\)</span></p></li>
</ul>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="technical-details">
<h1>Technical Details<a class="headerlink" href="#technical-details" title="Link to this heading">#</a></h1>
<section id="key-challenges-in-statistical-genetics">
<h2>Key Challenges in Statistical Genetics<a class="headerlink" href="#key-challenges-in-statistical-genetics" title="Link to this heading">#</a></h2>
<p>The application of LMMs in statistical genetics faces several major computational and methodological challenges that have driven the development of numerous specialized methods:</p>
<section id="computational-complexity">
<h3>Computational Complexity<a class="headerlink" href="#computational-complexity" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Matrix inversion of <span class="math notranslate nohighlight">\(\mathbf{V}^{-1}\)</span> requires <span class="math notranslate nohighlight">\(O(N^3)\)</span> operations for <span class="math notranslate nohighlight">\(N\)</span> individuals</p></li>
<li><p>Modern genomic datasets often have <span class="math notranslate nohighlight">\(N &gt; 100,000\)</span>, making direct inversion computationally prohibitive</p></li>
<li><p>Genetic relationship matrices are typically dense, exacerbating computational burden</p></li>
</ul>
</section>
<section id="high-dimensional-genomic-data">
<h3>High-Dimensional Genomic Data<a class="headerlink" href="#high-dimensional-genomic-data" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Genome-wide association studies (GWAS) test millions of variants</p></li>
<li><p>Each variant requires separate LMM fitting, multiplying computational demands</p></li>
<li><p>Standard software like <code class="docutils literal notranslate"><span class="pre">lmer</span></code> becomes impractical for genomic-scale analyses</p></li>
</ul>
</section>
<section id="complex-population-structure">
<h3>Complex Population Structure<a class="headerlink" href="#complex-population-structure" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Population stratification creates intricate correlation patterns</p></li>
<li><p>Cryptic relatedness between seemingly unrelated individuals</p></li>
<li><p>Multiple ancestral populations with different allele frequencies</p></li>
</ul>
</section>
</section>
<section id="why-many-specialized-methods-exist">
<h2>Why Many Specialized Methods Exist<a class="headerlink" href="#why-many-specialized-methods-exist" title="Link to this heading">#</a></h2>
<p>These challenges have necessitated the development of multiple specialized LMM implementations in statistical genetics, each optimizing different aspects:</p>
<section id="computational-efficiency-methods">
<h3>Computational Efficiency Methods<a class="headerlink" href="#computational-efficiency-methods" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>BOLT-LMM</strong> (Bayesian On-Line Learning Tool): Employs Monte Carlo methods and randomized algorithms for large-scale GWAS</p></li>
<li><p><strong>SAIGE</strong> (Scalable and Accurate Implementation of GEneralized mixed model): Uses sparse matrix techniques and saddlepoint approximation</p></li>
<li><p><strong>fastGWA</strong>: Implements sparse genetic relationship matrices and efficient algorithms for biobank-scale data</p></li>
</ul>
</section>
<section id="memory-efficient-approaches">
<h3>Memory-Efficient Approaches<a class="headerlink" href="#memory-efficient-approaches" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>GCTA</strong> (Genome-wide Complex Trait Analysis): Uses restricted maximum likelihood (REML) with memory-efficient algorithms</p></li>
<li><p><strong>LDAK</strong> (LD Adjusted Kinship): Adjusts for linkage disequilibrium patterns to improve kinship matrix estimation</p></li>
<li><p><strong>REGENIE</strong>: Two-step approach that first fits whole-genome model, then tests individual variants</p></li>
</ul>
</section>
<section id="statistical-power-optimization">
<h3>Statistical Power Optimization<a class="headerlink" href="#statistical-power-optimization" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Mixed Linear Model</strong> implementations in <strong>PLINK 2.0</strong>: Optimized for standard GWAS workflows</p></li>
<li><p><strong>EIGENSOFT/EIGENSTRAT</strong>: Principal component-based population structure correction</p></li>
</ul>
<p>This diversity of approaches reflects the fundamental tension between computational feasibility and statistical rigor in modern genomic analyses, where the scale of data has far exceeded traditional statistical computing paradigms. Each method represents different trade-offs between speed, memory usage, statistical accuracy, and applicability to specific study designs.</p>
</section>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="example">
<h1>Example<a class="headerlink" href="#example" title="Link to this heading">#</a></h1>
<p>In this genetic analysis example, we examine the relationship between genetic variants and height using <strong>random effect models</strong>.</p>
<p>The key steps include:</p>
<ol class="arabic simple">
<li><p>preparing genotype and phenotype data for 5 individuals with 3 genetic variants and height measurements, and standarize them</p></li>
<li><p>organizing individuals into family groups (individuals 1-3 in family 1, individuals 4-5 in family 2)</p></li>
<li><p>fitting random effect models for each variant using the <code class="docutils literal notranslate"><span class="pre">lmer</span></code> function, which allows us to separate variant-specific effects from family-related genetic background.</p></li>
</ol>
<p>The key information captured includes the effect size of each variant on height, the proportion of height variance explained by family relatedness versus the specific variant, and how accounting for family structure affects our understanding of genetic associations. Unlike fixed effect models that might attribute too much importance to specific variants, this approach helps control for the shared genetic background within families.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Clear the environment</span>
<span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ls</span><span class="p">())</span>

<span class="c1"># Load required packages</span>
<span class="nf">library</span><span class="p">(</span><span class="n">lme4</span><span class="p">)</span><span class="w">  </span><span class="c1"># For mixed-effect models</span>

<span class="c1"># Define genotypes for 5 individuals at 3 variants</span>
<span class="c1"># These represent actual alleles at each position</span>
<span class="c1"># For example, Individual 1 has genotypes: CC, CT, AT</span>
<span class="n">genotypes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span>
<span class="w"> </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AT&quot;</span><span class="p">,</span><span class="w">  </span><span class="c1"># Individual 1</span>
<span class="w"> </span><span class="s">&quot;TT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AA&quot;</span><span class="p">,</span><span class="w">  </span><span class="c1"># Individual 2</span>
<span class="w"> </span><span class="s">&quot;CT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AA&quot;</span><span class="p">,</span><span class="w">  </span><span class="c1"># Individual 3</span>
<span class="w"> </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AA&quot;</span><span class="p">,</span><span class="w">  </span><span class="c1"># Individual 4</span>
<span class="w"> </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TT&quot;</span><span class="w">   </span><span class="c1"># Individual 5</span>
<span class="p">)</span>
<span class="c1"># Reshape into a matrix</span>
<span class="n">N</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="c1"># number of individuals</span>
<span class="n">M</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="c1"># number of variants</span>
<span class="n">geno_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="n">genotypes</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="o">=</span><span class="n">N</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="n">M</span><span class="p">,</span><span class="w"> </span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">geno_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Individual&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">N</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">geno_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Variant&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span>

<span class="n">alt_alleles</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;T&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;C&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;T&quot;</span><span class="p">)</span>

<span class="c1"># Convert to raw genotype matrix using the additive / dominant / recessive model</span>
<span class="n">Xraw_additive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="o">=</span><span class="n">N</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="n">M</span><span class="p">)</span><span class="w"> </span><span class="c1"># dount number of non-reference alleles</span>

<span class="nf">rownames</span><span class="p">(</span><span class="n">Xraw_additive</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">geno_matrix</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">Xraw_additive</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">geno_matrix</span><span class="p">)</span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">N</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">alleles</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">strsplit</span><span class="p">(</span><span class="n">geno_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">],</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]]</span>
<span class="w">    </span><span class="n">Xraw_additive</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">alleles</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">alt_alleles</span><span class="p">[</span><span class="n">j</span><span class="p">])</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
<span class="n">X</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">scale</span><span class="p">(</span><span class="n">Xraw_additive</span><span class="p">,</span><span class="w"> </span><span class="n">center</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">scale</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># assign observed height for the 5 individuals</span>
<span class="n">Y_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">180</span><span class="p">,</span><span class="w"> </span><span class="m">160</span><span class="p">,</span><span class="w"> </span><span class="m">158</span><span class="p">,</span><span class="w"> </span><span class="m">155</span><span class="p">,</span><span class="w"> </span><span class="m">193</span><span class="p">)</span>
<span class="n">Y</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">scale</span><span class="p">(</span><span class="n">Y_raw</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Loading required package: Matrix
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Add family information (individuals 1,2,3 are from family 1, and 4,5 from family 2)</span>
<span class="n">family_info</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>

<span class="c1"># Create a data frame for analysis with scaled genotypes</span>
<span class="n">genetic_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">  </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Y</span><span class="p">,</span>
<span class="w">  </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">family_info</span><span class="p">,</span>
<span class="w">  </span><span class="n">individual</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">N</span><span class="p">,</span>
<span class="w">  </span><span class="n">X</span><span class="w">  </span><span class="c1"># This directly adds all columns of X to the dataframe</span>
<span class="p">)</span>

<span class="c1"># Rename the genotype columns for clarity</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">genetic_data</span><span class="p">)[</span><span class="m">4</span><span class="o">:</span><span class="p">(</span><span class="m">3</span><span class="o">+</span><span class="n">M</span><span class="p">)]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;variant&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span>

<span class="n">genetic_data</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 5 x 6</caption>
<thead>
	<tr><th></th><th scope=col>height</th><th scope=col>family</th><th scope=col>individual</th><th scope=col>variant1</th><th scope=col>variant2</th><th scope=col>variant3</th></tr>
	<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>Individual 1</th><td> 0.6528093</td><td>1</td><td>1</td><td>-0.6708204</td><td> 0.2390457</td><td> 0.4472136</td></tr>
	<tr><th scope=row>Individual 2</th><td>-0.5560968</td><td>1</td><td>2</td><td> 1.5652476</td><td>-0.9561829</td><td>-0.6708204</td></tr>
	<tr><th scope=row>Individual 3</th><td>-0.6769875</td><td>1</td><td>3</td><td> 0.4472136</td><td> 0.2390457</td><td>-0.6708204</td></tr>
	<tr><th scope=row>Individual 4</th><td>-0.8583234</td><td>2</td><td>4</td><td>-0.6708204</td><td>-0.9561829</td><td>-0.6708204</td></tr>
	<tr><th scope=row>Individual 5</th><td> 1.4385984</td><td>2</td><td>5</td><td>-0.6708204</td><td> 1.4342743</td><td> 1.5652476</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Single variant analysis and PVE results</span>
<span class="n">pve_results_single_variants</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">  </span><span class="n">Variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">character</span><span class="p">(),</span>
<span class="w">  </span><span class="n">PVE_Fixed</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">numeric</span><span class="p">(),</span>
<span class="w">  </span><span class="n">PVE_Family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">numeric</span><span class="p">(),</span>
<span class="w">  </span><span class="n">PVE_Residual</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">numeric</span><span class="p">(),</span>
<span class="w">  </span><span class="n">Fixed_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">numeric</span><span class="p">(),</span>
<span class="w">  </span><span class="n">Family_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">numeric</span><span class="p">(),</span>
<span class="w">  </span><span class="n">Residual_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">numeric</span><span class="p">(),</span>
<span class="w">  </span><span class="n">Total_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">numeric</span><span class="p">(),</span>
<span class="w">  </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span>
<span class="p">)</span>

<span class="c1"># Calculate PVE for each variant individually</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">variant_col</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;variant&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">)</span>
<span class="w">  </span><span class="n">formula</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.formula</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;height ~&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_col</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;+ (1|family)&quot;</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Fit random effect model</span>
<span class="w">  </span><span class="n">model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lmer</span><span class="p">(</span><span class="n">formula</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genetic_data</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Extract variance components</span>
<span class="w">  </span><span class="n">vc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">VarCorr</span><span class="p">(</span><span class="n">model</span><span class="p">)</span>
<span class="w">  </span><span class="n">family_variance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">vc</span><span class="o">$</span><span class="n">family</span><span class="p">)</span>
<span class="w">  </span><span class="n">residual_variance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">attr</span><span class="p">(</span><span class="n">vc</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sc&quot;</span><span class="p">)</span><span class="o">^</span><span class="m">2</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Calculate fixed effect variance</span>
<span class="w">  </span><span class="n">beta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fixef</span><span class="p">(</span><span class="n">model</span><span class="p">)[</span><span class="n">variant_col</span><span class="p">]</span>
<span class="w">  </span><span class="n">X_var</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="n">genetic_data</span><span class="p">[[</span><span class="n">variant_col</span><span class="p">]])</span>
<span class="w">  </span><span class="n">fixed_variance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">var</span><span class="p">(</span><span class="nf">as.vector</span><span class="p">(</span><span class="n">X_var</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">beta</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Calculate total variance</span>
<span class="w">  </span><span class="n">total_variance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fixed_variance</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">family_variance</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">residual_variance</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Calculate PVE for each component</span>
<span class="w">  </span><span class="n">pve_fixed</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fixed_variance</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_variance</span>
<span class="w">  </span><span class="n">pve_family</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">family_variance</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_variance</span>
<span class="w">  </span><span class="n">pve_residual</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">residual_variance</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_variance</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Store results in the data frame</span>
<span class="w">  </span><span class="n">pve_results_single_variants</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">pve_results_single_variants</span><span class="p">,</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">    </span><span class="n">Variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variant_col</span><span class="p">,</span>
<span class="w">    </span><span class="n">PVE_Fixed</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pve_fixed</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span>
<span class="w">    </span><span class="n">PVE_Family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pve_family</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span>
<span class="w">    </span><span class="n">PVE_Residual</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pve_residual</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span>
<span class="w">    </span><span class="n">Fixed_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fixed_variance</span><span class="p">,</span>
<span class="w">    </span><span class="n">Family_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">family_variance</span><span class="p">,</span>
<span class="w">    </span><span class="n">Residual_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">residual_variance</span><span class="p">,</span>
<span class="w">    </span><span class="n">Total_Variance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total_variance</span><span class="p">,</span>
<span class="w">    </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span>
<span class="w">  </span><span class="p">))</span>
<span class="p">}</span>

<span class="n">pve_results_single_variants</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>boundary (singular) fit: see help(&#39;isSingular&#39;)

boundary (singular) fit: see help(&#39;isSingular&#39;)
</pre></div>
</div>
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 3 x 8</caption>
<thead>
	<tr><th scope=col>Variant</th><th scope=col>PVE_Fixed</th><th scope=col>PVE_Family</th><th scope=col>PVE_Residual</th><th scope=col>Fixed_Variance</th><th scope=col>Family_Variance</th><th scope=col>Residual_Variance</th><th scope=col>Total_Variance</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>variant1</td><td>20.00779</td><td>0.000000</td><td>79.9922054</td><td>0.2500913</td><td>0.00000000</td><td>0.999878212</td><td>1.24997</td></tr>
	<tr><td>variant2</td><td>66.60926</td><td>0.000000</td><td>33.3907386</td><td>0.7267603</td><td>0.00000000</td><td>0.364319641</td><td>1.09108</td></tr>
	<tr><td>variant3</td><td>95.70867</td><td>3.655575</td><td> 0.6357522</td><td>1.1000658</td><td>0.04201681</td><td>0.007307271</td><td>1.14939</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Calculate PVE for all variants together</span>
<span class="c1"># Fit the full model with all variants</span>
<span class="n">full_formula</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.formula</span><span class="p">(</span><span class="s">&quot;height ~ variant1 + variant2 + variant3 + (1|family)&quot;</span><span class="p">)</span>
<span class="n">full_model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lmer</span><span class="p">(</span><span class="n">full_formula</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genetic_data</span><span class="p">)</span>

<span class="c1"># Extract variance components</span>
<span class="n">vc_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">VarCorr</span><span class="p">(</span><span class="n">full_model</span><span class="p">)</span>
<span class="n">family_variance_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">vc_full</span><span class="o">$</span><span class="n">family</span><span class="p">)</span>
<span class="n">residual_variance_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">attr</span><span class="p">(</span><span class="n">vc_full</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sc&quot;</span><span class="p">)</span><span class="o">^</span><span class="m">2</span>

<span class="c1"># Calculate fixed effect variance for the full model</span>
<span class="n">betas</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fixef</span><span class="p">(</span><span class="n">full_model</span><span class="p">)[</span><span class="m">2</span><span class="o">:</span><span class="p">(</span><span class="n">M</span><span class="m">+1</span><span class="p">)]</span><span class="w">  </span><span class="c1"># Skip intercept</span>
<span class="n">X_design</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">genetic_data</span><span class="p">[,</span><span class="w"> </span><span class="m">4</span><span class="o">:</span><span class="p">(</span><span class="m">3</span><span class="o">+</span><span class="n">M</span><span class="p">)])</span>
<span class="n">predicted_fixed</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">X_design</span><span class="w"> </span><span class="o">%*%</span><span class="w"> </span><span class="n">betas</span>
<span class="n">fixed_variance_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">var</span><span class="p">(</span><span class="nf">as.vector</span><span class="p">(</span><span class="n">predicted_fixed</span><span class="p">))</span>

<span class="c1"># Calculate total variance</span>
<span class="n">total_variance_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fixed_variance_full</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">family_variance_full</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">residual_variance_full</span>

<span class="c1"># Calculate PVE for each component</span>
<span class="n">pve_fixed_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fixed_variance_full</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_variance_full</span>
<span class="n">pve_family_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">family_variance_full</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_variance_full</span>
<span class="n">pve_residual_full</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">residual_variance_full</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_variance_full</span>

<span class="c1"># Print out the results for all variants together</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;\n---------- PVE Analysis for All Variants Together ----------\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Fixed effects (all variants) PVE:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">pve_fixed_full</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;%\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Random effect (family) PVE:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">pve_family_full</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;%\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Residual PVE:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">pve_residual_full</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;%\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Total:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">pve_fixed_full</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">pve_family_full</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">pve_residual_full</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;%\n\n&quot;</span><span class="p">)</span>

<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Variance components:\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Fixed effects variance:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">fixed_variance_full</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Family variance:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">family_variance_full</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Residual variance:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">residual_variance_full</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Total variance:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">total_variance_full</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>

<span class="c1"># Calculate heritability (proportion of variance due to family structure)</span>
<span class="n">heritability</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">family_variance_full</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="p">(</span><span class="n">family_variance_full</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">residual_variance_full</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;\nHeritability (proportion of variance due to family structure):&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">    </span><span class="nf">round</span><span class="p">(</span><span class="n">heritability</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;%\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in checkConv(attr(opt, &quot;derivs&quot;), opt$par, ctrl = control$checkConv, :
&quot;unable to evaluate scaled gradient&quot;
Warning message in checkConv(attr(opt, &quot;derivs&quot;), opt$par, ctrl = control$checkConv, :
&quot; Hessian is numerically singular: parameters are not uniquely determined&quot;
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>---------- PVE Analysis for All Variants Together ----------
Fixed effects (all variants) PVE: 92.48 %
Random effect (family) PVE: 1.41 %
Residual PVE: 6.11 %
Total: 100 %

Variance components:
Fixed effects variance: 1.0142 
Family variance: 0.0155 
Residual variance: 0.067 
Total variance: 1.0967 

Heritability (proportion of variance due to family structure): 18.77 %
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Perform random effect model analysis for each variant separately</span>
<span class="n">random_models</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>
<span class="n">random_model_summaries</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">variant_col</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;variant&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">)</span>
<span class="w">  </span><span class="n">formula</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.formula</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;height ~&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_col</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;+ (1|family)&quot;</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Fit random effect model</span>
<span class="w">  </span><span class="n">random_models</span><span class="p">[[</span><span class="n">j</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lmer</span><span class="p">(</span><span class="n">formula</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genetic_data</span><span class="p">)</span>
<span class="w">  </span><span class="n">random_model_summaries</span><span class="p">[[</span><span class="n">j</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">summary</span><span class="p">(</span><span class="n">random_models</span><span class="p">[[</span><span class="n">j</span><span class="p">]])</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Extract and print key information</span>
<span class="w">  </span><span class="n">fixed_effects</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fixef</span><span class="p">(</span><span class="n">random_models</span><span class="p">[[</span><span class="n">j</span><span class="p">]])</span>
<span class="w">  </span><span class="n">random_variance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">VarCorr</span><span class="p">(</span><span class="n">random_models</span><span class="p">[[</span><span class="n">j</span><span class="p">]])</span><span class="o">$</span><span class="n">family</span><span class="p">)</span>
<span class="w">  </span><span class="n">residual_variance</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">attr</span><span class="p">(</span><span class="nf">VarCorr</span><span class="p">(</span><span class="n">random_models</span><span class="p">[[</span><span class="n">j</span><span class="p">]]),</span><span class="w"> </span><span class="s">&quot;sc&quot;</span><span class="p">)</span><span class="o">^</span><span class="m">2</span>
<span class="w">  </span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;\nEffect size (beta):&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fixed_effects</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Family variance component:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">random_variance</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Residual variance component:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">residual_variance</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Proportion of variance due to family:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">random_variance</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="p">(</span><span class="n">random_variance</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">residual_variance</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>boundary (singular) fit: see help(&#39;isSingular&#39;)
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Effect size (beta): -0.5000913 
Family variance component: 0 
Residual variance component: 0.9998782 
Proportion of variance due to family: 0 
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>boundary (singular) fit: see help(&#39;isSingular&#39;)
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Effect size (beta): 0.8525024 
Family variance component: 0 
Residual variance component: 0.3643196 
Proportion of variance due to family: 0 

Effect size (beta): 1.04884 
Family variance component: 0.04201681 
Residual variance component: 0.007307271 
Proportion of variance due to family: 0.8518519 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Compare effects across variants</span>
<span class="n">variant_effects</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="n">random_models</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">model</span><span class="p">)</span><span class="w"> </span><span class="nf">fixef</span><span class="p">(</span><span class="n">model</span><span class="p">)[</span><span class="m">2</span><span class="p">])</span>
<span class="nf">names</span><span class="p">(</span><span class="n">variant_effects</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;Variant &quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;\n---------- Effect Sizes Across Variants ----------\n&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">variant_effects</span><span class="p">)</span>

<span class="c1"># Compare variance components across variants</span>
<span class="n">variance_components</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lapply</span><span class="p">(</span><span class="n">random_models</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">model</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">vc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">VarCorr</span><span class="p">(</span><span class="n">model</span><span class="p">)</span>
<span class="w">  </span><span class="n">family_var</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">vc</span><span class="o">$</span><span class="n">family</span><span class="p">)</span>
<span class="w">  </span><span class="n">residual_var</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">attr</span><span class="p">(</span><span class="n">vc</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sc&quot;</span><span class="p">)</span><span class="o">^</span><span class="m">2</span>
<span class="w">  </span><span class="nf">c</span><span class="p">(</span><span class="n">family_var</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">family_var</span><span class="p">,</span><span class="w"> </span>
<span class="w">    </span><span class="n">residual_var</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">residual_var</span><span class="p">,</span><span class="w"> </span>
<span class="w">    </span><span class="n">proportion_family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">family_var</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="p">(</span><span class="n">family_var</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">residual_var</span><span class="p">))</span>
<span class="p">})</span>
<span class="n">variance_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">do.call</span><span class="p">(</span><span class="n">rbind</span><span class="p">,</span><span class="w"> </span><span class="n">variance_components</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">variance_df</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;Variant &quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;\n---------- Variance Components Across Variants ----------\n&quot;</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">variance_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>---------- Effect Sizes Across Variants ----------
 Variant 1  Variant 2  Variant 3 
-0.5000913  0.8525024  1.0488402 

---------- Variance Components Across Variants ----------
          family_var residual_var proportion_family
Variant 1 0.00000000  0.999878212         0.0000000
Variant 2 0.00000000  0.364319641         0.0000000
Variant 3 0.04201681  0.007307271         0.8518519
</pre></div>
</div>
</div>
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="proportion_of_variance_explained.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Proportion of Variance Explained and Heritability</p>
      </div>
    </a>
    <a class="right-next"
       href="confounder.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Confounder</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Linear Mixed Model</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#graphical-summary">Graphical Summary</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#key-formula">Key Formula</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#technical-details">Technical Details</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#key-challenges-in-statistical-genetics">Key Challenges in Statistical Genetics</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#computational-complexity">Computational Complexity</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#high-dimensional-genomic-data">High-Dimensional Genomic Data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#complex-population-structure">Complex Population Structure</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#why-many-specialized-methods-exist">Why Many Specialized Methods Exist</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#computational-efficiency-methods">Computational Efficiency Methods</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#memory-efficient-approaches">Memory-Efficient Approaches</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#statistical-power-optimization">Statistical Power Optimization</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#example">Example</a></li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Gao Wang
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      Â© Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>