<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Maximum Likelihood Estimation &#8212; Statistical backgrounds for the Summer Program in Statistical Functional Genomics</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=12dfc556" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
    <script src="_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/scripts/sphinx-book-theme.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Likelihood Ratio" href="likelihood_ratio.html" />
    <link rel="prev" title="Likelihood" href="likelihood.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section class="tex2jax_ignore mathjax_ignore" id="maximum-likelihood-estimation">
<h1>Maximum Likelihood Estimation<a class="headerlink" href="#maximum-likelihood-estimation" title="Link to this heading">¶</a></h1>
<p>Maximum likelihood estimation identifies the parameter values that maximize the probability of observing your actual data, essentially asking “which model settings would make what we actually saw most likely to occur?”</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="graphical-summary">
<h1>Graphical Summary<a class="headerlink" href="#graphical-summary" title="Link to this heading">¶</a></h1>
<p><img alt="fig" src="_images/maximum_likelihood_estimation.svg" /></p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="key-formula">
<h1>Key Formula<a class="headerlink" href="#key-formula" title="Link to this heading">¶</a></h1>
<p>In statistics, maximum likelihood estimation (MLE) is a method of estimating the parameters of an assumed probability distribution, given some observed data. This is achieved by finding the values of the model parameters (e.g., <span class="math notranslate nohighlight">\(\theta\)</span>) that maximize the likelihood function (<span class="math notranslate nohighlight">\(L(\theta|\text{D}) = P(\theta|\text{D})\)</span>) over the parameter space (e.g., <span class="math notranslate nohighlight">\(\Theta\)</span>), so that, under the assumed statistical model, the observed data <span class="math notranslate nohighlight">\(\text{D}\)</span> is most probable.</p>
<div class="math notranslate nohighlight">
\[
\hat{\theta} = \underset{\theta \in \Theta}{\operatorname{argmax}} L(\theta|D) = \underset{\theta \in \Theta}{\operatorname{argmax}} P(D|\theta)
\]</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="technical-details">
<h1>Technical Details<a class="headerlink" href="#technical-details" title="Link to this heading">¶</a></h1>
<section id="work-with-log-likelihood">
<h2>Work with Log-Likelihood<a class="headerlink" href="#work-with-log-likelihood" title="Link to this heading">¶</a></h2>
<p>In practice, it is often convenient to work with the natural logarithm of the likelihood function, called the log-likelihood:</p>
<div class="math notranslate nohighlight">
\[
\ell (\theta \,;\mathbf{y})=\ln {\mathcal{L}}_{n}(\theta \,;\mathbf{y})
\]</div>
<p>Since the logarithm is a monotonic function, the maximum of <span class="math notranslate nohighlight">\(\ell (\theta \,;\mathbf{y})\)</span> occurs at the same value of <span class="math notranslate nohighlight">\(\theta\)</span> as does the maximum of <span class="math notranslate nohighlight">\({\mathcal{L}}_{n}\)</span>.</p>
</section>
<section id="likelihood-equation">
<h2>Likelihood equation<a class="headerlink" href="#likelihood-equation" title="Link to this heading">¶</a></h2>
<p>If <span class="math notranslate nohighlight">\(\ell (\theta \,;\mathbf{y})\)</span> is differentiable in <span class="math notranslate nohighlight">\(\Theta\)</span>, sufficient conditions for the occurrence of a maximum (or a minimum) are</p>
<div class="math notranslate nohighlight">
\[
\frac{\partial \ell}{\partial \theta}=0
\]</div>
<p>known as the likelihood equations.</p>
</section>
<section id="iid-data">
<h2>IID data<a class="headerlink" href="#iid-data" title="Link to this heading">¶</a></h2>
<p>If the data are independent and identically distributed, then we have</p>
<div class="math notranslate nohighlight">
\[
{\widehat{\ell\,}}(\theta \,;D)=\sum_{i=1}^{n}\ln f(d_{i}\mid \theta)
\]</div>
<p>this being the sample analogue of the expected log-likelihood</p>
<div class="math notranslate nohighlight">
\[
\ell(\theta)=\operatorname{\mathbb{E}}[\,\ln f(d_{i}\mid \theta)\,]
\]</div>
<p>where this expectation is taken with respect to the true density.</p>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="example">
<h1>Example<a class="headerlink" href="#example" title="Link to this heading">¶</a></h1>
<p>Here we use the first genetic variant to show how to calculate the maximum likelihood estimate of the frequency of a genetic variant, using the first genetic variant as an example.</p>
<section id="obtain-the-mle-for-frequency-from-the-figure">
<h2>Obtain the MLE for frequency from the figure<a class="headerlink" href="#obtain-the-mle-for-frequency-from-the-figure" title="Link to this heading">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Clear the environment</span>
<span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ls</span><span class="p">())</span>

<span class="c1"># Define genotypes for 5 individuals at 3 variants</span>
<span class="c1"># These represent actual alleles at each position</span>
<span class="c1"># For example, Individual 1 has genotypes: CC, CT, AT</span>
<span class="n">genotypes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span>
<span class="w">    </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AT&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># Individual 1</span>
<span class="w">    </span><span class="s">&quot;TT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AA&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># Individual 2</span>
<span class="w">    </span><span class="s">&quot;CT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AA&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># Individual 3</span>
<span class="w">    </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TT&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AA&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># Individual 4</span>
<span class="w">    </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CC&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TT&quot;</span><span class="w"> </span><span class="c1"># Individual 5</span>
<span class="p">)</span>
<span class="c1"># Reshape into a matrix</span>
<span class="n">N</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="c1"># number of individuals</span>
<span class="n">M</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="c1"># number of variants</span>
<span class="n">geno_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="n">genotypes</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">N</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">M</span><span class="p">,</span><span class="w"> </span><span class="n">byrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">geno_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Individual&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">N</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">geno_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Variant&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">M</span><span class="p">)</span>

<span class="n">alt_alleles</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;T&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;C&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;T&quot;</span><span class="p">)</span>
<span class="n">ref_alleles</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;C&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;T&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;A&quot;</span><span class="p">)</span>

<span class="n">geno_matrix</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A matrix: 5 × 3 of type chr</caption>
<thead>
	<tr><th></th><th scope=col>Variant 1</th><th scope=col>Variant 2</th><th scope=col>Variant 3</th></tr>
</thead>
<tbody>
	<tr><th scope=row>Individual 1</th><td>CC</td><td>CT</td><td>AT</td></tr>
	<tr><th scope=row>Individual 2</th><td>TT</td><td>TT</td><td>AA</td></tr>
	<tr><th scope=row>Individual 3</th><td>CT</td><td>CT</td><td>AA</td></tr>
	<tr><th scope=row>Individual 4</th><td>CC</td><td>TT</td><td>AA</td></tr>
	<tr><th scope=row>Individual 5</th><td>CC</td><td>CC</td><td>TT</td></tr>
</tbody>
</table>
</div></div>
</div>
<p>In the first variant we have observed the genotype data for five individuals, CC, TT, CT, CC and CC (which is our data <span class="math notranslate nohighlight">\(\text{D}\)</span>). We want to estimate the frequency fo the alternative <code class="docutils literal notranslate"><span class="pre">T</span></code> for this variant. Specifically, for each <span class="math notranslate nohighlight">\(q\in[0,1]\)</span> let <span class="math notranslate nohighlight">\(M_q\)</span> denote the model that the true frequency of the <code class="docutils literal notranslate"><span class="pre">T</span></code> allele is <span class="math notranslate nohighlight">\(q\)</span>. Then, given our observation that 3 of 10 alleles (2 alleles per individual because human is dipliod) carried allele <code class="docutils literal notranslate"><span class="pre">T</span></code> at marker 1, the likelihood for model <span class="math notranslate nohighlight">\(M_q\)</span> is, by the previous definition,
$<span class="math notranslate nohighlight">\(
L(M_q)=\Pr(\text{D}|M_q)=(1-q)^2 * q^2 * 2q(1-q) * (1-q)^2 * (1-q)^2 = 2q^3(1−q)^7
\)</span>$</p>
<p>When computing likelihoods for parametric models, we usually dispense with the model notation and simply use the parameter value to denote the model. So instead of referring to the likelihood for <span class="math notranslate nohighlight">\(M_q\)</span>  we just say the “likelihood for <span class="math notranslate nohighlight">\(q\)</span>”, and write <span class="math notranslate nohighlight">\(L(q)\)</span>. So the likelihood for <span class="math notranslate nohighlight">\(q\)</span> is given by</p>
<div class="math notranslate nohighlight">
\[
L(q)= 2q^3(1−q)^7
\]</div>
<p>We could plot the likelihood function as follows:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">q</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">seq</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="n">L</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">q</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">q</span><span class="o">^</span><span class="m">3</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="m">1</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">q</span><span class="p">)</span><span class="o">^</span><span class="m">7</span>
<span class="p">}</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">q</span><span class="p">,</span><span class="w"> </span><span class="nf">L</span><span class="p">(</span><span class="n">q</span><span class="p">),</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;L(q)&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;q&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;l&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/a138eb3b90422e8dc6afda035694e4178b324dedcee5b2c6370624df54927e29.png" src="_images/a138eb3b90422e8dc6afda035694e4178b324dedcee5b2c6370624df54927e29.png" />
</div>
</div>
<p>Since the value of <span class="math notranslate nohighlight">\(q\)</span> that maximizes the likelihood function is referred to as the “maximum likelihood estimate”, and usually denoted <span class="math notranslate nohighlight">\(\hat{q}\)</span>. That is
$<span class="math notranslate nohighlight">\(
\hat{q} =\underset{\theta \in \Theta}{\operatorname{argmax}} L(q)
\)</span>$</p>
<p>Provided the data are sufficiently informative, and the number of parameters is not too large, maximum likelihood estimates tend to be sensible. In this case we can see that the maximum likelihood estimate is <span class="math notranslate nohighlight">\(q=0.3\)</span>, which also corresponds to our intuition.</p>
</section>
<section id="from-likelihood-function-to-log-likelihood-function">
<h2>From likelihood function to log-likelihood function<a class="headerlink" href="#from-likelihood-function-to-log-likelihood-function" title="Link to this heading">¶</a></h2>
<p>Just as it can often be convenient to work with the log-likelihood ratio, it can be convenient to work with the log-likelihood function, usually denoted as <span class="math notranslate nohighlight">\(l(q)\)</span>.  Here is the log-likelihood function.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">q</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">seq</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="n">l</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">q</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="m">3</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">log</span><span class="p">(</span><span class="n">q</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">7</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">log</span><span class="p">(</span><span class="m">1</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">q</span><span class="p">)</span>
<span class="p">}</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">q</span><span class="p">,</span><span class="w"> </span><span class="nf">l</span><span class="p">(</span><span class="n">q</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">l</span><span class="p">(</span><span class="m">0.3</span><span class="p">),</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;l(q) - l(qhat)&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;q&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;l&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ylim</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-10</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/10976827333bebdbca4008974e44d277493a00cf19a6fb960cea19d9264ed674.png" src="_images/10976827333bebdbca4008974e44d277493a00cf19a6fb960cea19d9264ed674.png" />
</div>
</div>
<p>Changes in the log-likelihood function are referred to as “log-likelihood units”. For example the difference in the support for <code class="docutils literal notranslate"><span class="pre">q=0.3</span></code> and <code class="docutils literal notranslate"><span class="pre">q=0.35</span></code> is <code class="docutils literal notranslate"><span class="pre">l(0.3)-l(0.35)</span> <span class="pre">=</span> <span class="pre">0.5630377</span></code> log-likelihood units. Again, remember that it is differences in <code class="docutils literal notranslate"><span class="pre">l</span></code> that matter, not the actual values.</p>
</section>
<section id="obtain-the-mle-of-frequency-analytically">
<h2>Obtain the MLE of frequency analytically<a class="headerlink" href="#obtain-the-mle-of-frequency-analytically" title="Link to this heading">¶</a></h2>
<p>In this simple case we can calculate <span class="math notranslate nohighlight">\(\hat{q}\)</span> analytically as well, but keep in mind that in most of the cases, it’s impossible to get the explicit formula of <span class="math notranslate nohighlight">\(\hat{\theta}\)</span> of a parameter <span class="math notranslate nohighlight">\(\theta\)</span> directly.</p>
<ol class="arabic">
<li><p>Take the natural logarithm of the likelihood function (for computational convenience):
$<span class="math notranslate nohighlight">\(\ln L(q) = \ln(2) + 3\ln(q) + 7\ln(1-q)\)</span>$</p></li>
<li><p>Find the critical point by taking the derivative and setting it to zero:
$<span class="math notranslate nohighlight">\(\frac{d\ln L(q)}{dq} = \frac{3}{q} - \frac{7}{1-q} = 0\)</span>$</p></li>
<li><p>Solve for <span class="math notranslate nohighlight">\(q\)</span>:
$<span class="math notranslate nohighlight">\(\frac{3}{q} = \frac{7}{1-q}\)</span><span class="math notranslate nohighlight">\(
\)</span><span class="math notranslate nohighlight">\(3(1-q) = 7q\)</span><span class="math notranslate nohighlight">\(
\)</span><span class="math notranslate nohighlight">\(3 - 3q = 7q\)</span><span class="math notranslate nohighlight">\(
\)</span><span class="math notranslate nohighlight">\(3 = 10q\)</span><span class="math notranslate nohighlight">\(
\)</span><span class="math notranslate nohighlight">\(q = \frac{3}{10} = 0.3\)</span>$</p></li>
<li><p>Verify it’s a maximum by checking that the second derivative is negative:
$<span class="math notranslate nohighlight">\(\frac{d^2\ln L(q)}{dq^2} = -\frac{3}{q^2} - \frac{7}{(1-q)^2} &lt; 0\)</span>$</p>
<p>This is always negative for <span class="math notranslate nohighlight">\(q \in (0,1)\)</span>, confirming we have a maximum.</p>
</li>
</ol>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="README.html">Project name not set</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Notations and References</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Notations.html"><strong>Notation Table</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="Notations.html#concepts-in-statistical-genetics-and-related-topics">Concepts in statistical genetics and related topics</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Basic Concepts</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html">Genotype Coding</a></li>
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html">Minor Allele Frequency</a></li>
<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html">Hardy-Weinberg Equilibrium</a></li>
<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html#example">Example</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Correlation Between Variants</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html">Linkage Disequilibrium</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html">Linkage Disequilibrium Score</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html#example">Example</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Correlation Between Individuals</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html">Genetic Relationship Matrix</a></li>
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html#example">Example</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Regression Models</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html">Ordinary Least Squares</a></li>
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html">Summary Statistics</a></li>
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html#todo">TODO</a></li>
<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html">Odds Ratio</a></li>
<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html">Linear Mixed Model</a></li>
<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html">Proportion of Variance Explained and Heritability</a></li>
<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html">Marginal and Joint Effects</a></li>
<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html#example">Example</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Covariates in Genetic Studies</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="confounder.html">Confounder</a></li>
<li class="toctree-l1"><a class="reference internal" href="confounder.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="confounder.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="confounder.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="confounder.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="collider.html">Collider</a></li>
<li class="toctree-l1"><a class="reference internal" href="collider.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="collider.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="collider.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="collider.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="mediator.html">Mediator</a></li>
<li class="toctree-l1"><a class="reference internal" href="mediator.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="mediator.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="mediator.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="mediator.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="mediator.html#supplementary">Supplementary</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">From One Study to Multiple Studies</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html">Meta Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html#supplementary">Supplementary</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Statistical Inference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="likelihood.html">Likelihood</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood.html#supplementary-materials">Supplementary Materials</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood.html#recommended-reading">Recommended Reading</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Maximum Likelihood Estimation</a></li>
<li class="toctree-l1"><a class="reference internal" href="#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html">Likelihood Ratio</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html#supplementary-materials">Supplementary Materials</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html#recommended-reading">Recommended Reading</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html">Likelihood Ratio Test</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html#supplementary-materials">Supplementary Materials</a></li>
<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html#recommended-reading">Recommended Reading</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Bayes Factor</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html">Bayes Factor</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="p_value_versus_Bayes_factor.html">P-value vs. Bayes Factor</a></li>
<li class="toctree-l1"><a class="reference internal" href="p_value_versus_Bayes_factor.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="p_value_versus_Bayes_factor.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="p_value_versus_Bayes_factor.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="p_value_versus_Bayes_factor.html#example">Example</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Bayes Models</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html">Bayesian Normal Mean Model</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html#supplementary">Supplementary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html">Bayesian Multivariate Normal Mean Model</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html">Bayesian Mixture Model</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html#example">Example</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Model Comparison and Averaging</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html">Bayesian Model Comparison</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html#example">Example</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_averaging.html">Bayesian Model Averaging</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_averaging.html#graphical-summary">Graphical Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_averaging.html#key-formula">Key Formula</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_averaging.html#technical-details">Technical Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_averaging.html#example">Example</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="README.html">Documentation overview</a><ul>
      <li>Previous: <a href="likelihood.html" title="previous chapter">Likelihood</a></li>
      <li>Next: <a href="likelihood_ratio.html" title="next chapter">Likelihood Ratio</a></li>
  </ul></li>
</ul>
</div>
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 7.4.7</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="_sources/maximum_likelihood_estimation.ipynb"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>