
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Notations &#8212; Statistical backgrounds for gene mapping</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notations';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Popular Topics in Statistical Genetics" href="popular_topics.html" />
    <link rel="prev" title="statgen-prerequisites" href="README.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="README.html">
  
  
  
  
  
  
    <p class="title logo__title">Statistical backgrounds for gene mapping</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Popular Topics and Related Notebooks</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Notations</a></li>
<li class="toctree-l1"><a class="reference internal" href="popular_topics.html">Popular Topics in Statistical Genetics</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Basic Concepts</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html">Genotype Coding</a></li>






<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html">Minor Allele Frequency</a></li>





<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html">Hardy-Weinberg Equilibrium</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Correlation Between Variants</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html">Linkage Disequilibrium</a></li>






<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html">Linkage Disequilibrium Score</a></li>






</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Correlation Between Individuals</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html">Genetic Relationship Matrix</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Fixed Effects Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html">Ordinary Least Squares</a></li>






<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html">Odds Ratio</a></li>






<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html">Marginal and Joint Effects</a></li>






<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html">Summary Statistics</a></li>






</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Random Effects Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="random_effect.html">Random Effect</a></li>





<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html">Proportion of Variance Explained and Heritability</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Linear Mixed Model</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html">Linear Mixed Model</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Covariates</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="confounder.html">Confounder</a></li>





<li class="toctree-l1"><a class="reference internal" href="collider.html">Collider</a></li>




<li class="toctree-l1"><a class="reference internal" href="mediator.html">Mediator</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Meta-analysis</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="meta_analysis_fixed_effect.html">Meta-Analysis Fixed Effect</a></li>





<li class="toctree-l1"><a class="reference internal" href="meta_analysis_random_effect.html">Meta-Analysis Random Effect</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Likelihood</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="likelihood.html">Likelihood</a></li>






<li class="toctree-l1"><a class="reference internal" href="maximum_likelihood_estimation.html">Maximum Likelihood Estimation</a></li>






<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html">Likelihood Ratio</a></li>






</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Bayesian Versus Frequentist</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayesian_frequentist.html">Bayesian and Frequentist</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bayes_rule.html">Bayes Rule</a></li>






<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html">Bayes Factor</a></li>





<li class="toctree-l1"><a class="reference internal" href="p_value.html">p-value</a></li>






</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Bayesian Model</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html">Bayesian Normal Mean Model</a></li>






<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html">Bayesian Multivariate Normal Mean Model</a></li>






</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Multiple Bayesian models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html">Bayesian Model Comparison</a></li>






<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html">Bayesian Mixture Model</a></li>






</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/notations.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>

</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Notations</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="notations">
<h1>Notations<a class="headerlink" href="#notations" title="Link to this heading">#</a></h1>
<p>Below is a table of notations that we use in this course.</p>
<div class="pst-scrollable-table-container"><table class="table">
<thead>
<tr class="row-odd"><th class="head"><p>Symbol</p></th>
<th class="head"><p>Meaning</p></th>
<th class="head"><p>Example Meaning</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><strong>Genotype Data</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\mathbf{X}_{\text{raw}}\)</span></p></td>
<td><p>Raw genotype matrix (without centering or standardization)</p></td>
<td><p><span class="math notranslate nohighlight">\(N \times M\)</span> matrix with values 0, 1, 2 for AA, Aa, aa</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathbf{X}\)</span></p></td>
<td><p>Genotype matrix after normalization</p></td>
<td><p>Centered and scaled genotype matrix</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\mathbf{X}_{i,\cdot}\)</span></p></td>
<td><p>Genotype for individual <span class="math notranslate nohighlight">\(i\)</span> across all variants</p></td>
<td><p>Row vector of genotypes for individual <span class="math notranslate nohighlight">\(i\)</span></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathbf{X}_{i,j}\)</span></p></td>
<td><p>Genotype for individual <span class="math notranslate nohighlight">\(i\)</span> at variant <span class="math notranslate nohighlight">\(j\)</span></p></td>
<td><p>Number of risk alleles (0, 1, or 2) for individual <span class="math notranslate nohighlight">\(i\)</span> at variant <span class="math notranslate nohighlight">\(j\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\mathbf{X}_{\cdot,j}\)</span></p></td>
<td><p>Genotype at variant <span class="math notranslate nohighlight">\(j\)</span> for all individuals</p></td>
<td><p>Column vector of genotypes for variant <span class="math notranslate nohighlight">\(j\)</span> across all individuals</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(X_i\)</span></p></td>
<td><p>Genotype value for individual <span class="math notranslate nohighlight">\(i\)</span> (single variant case)</p></td>
<td><p>Number of risk alleles for individual <span class="math notranslate nohighlight">\(i\)</span> (0, 1, or 2)</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\mu_j\)</span></p></td>
<td><p>Mean value of variant <span class="math notranslate nohighlight">\(j\)</span> in <span class="math notranslate nohighlight">\(\mathbf{X}_{\text{raw}}\)</span></p></td>
<td><p>Mean genotype value across all individuals for variant <span class="math notranslate nohighlight">\(j\)</span></p></td>
</tr>
<tr class="row-even"><td><p><strong>Indices and Dimensions</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(i\)</span></p></td>
<td><p>Index for individual</p></td>
<td><p><span class="math notranslate nohighlight">\(i=1,2,...,N\)</span></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(j\)</span></p></td>
<td><p>Index for genetic variant</p></td>
<td><p><span class="math notranslate nohighlight">\(j=1,2,...,M\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(k\)</span></p></td>
<td><p>Index for study</p></td>
<td><p><span class="math notranslate nohighlight">\(k=1,2,...,K\)</span></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(c\)</span></p></td>
<td><p>Index for mixture component</p></td>
<td><p><span class="math notranslate nohighlight">\(c=1,2,...,C\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(N\)</span></p></td>
<td><p>Number of individuals</p></td>
<td><p>500,000 individuals in UK Biobank subset</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(M\)</span></p></td>
<td><p>Number of variants</p></td>
<td><p>1,000,000 SNPs after quality control</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(R\)</span></p></td>
<td><p>Number of random effects</p></td>
<td><p>5 random effects from gene 1 to gene 5</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(K\)</span></p></td>
<td><p>Number of studies</p></td>
<td><p>3 studies in meta-analysis</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(C\)</span></p></td>
<td><p>Number of mixture components</p></td>
<td><p>4 components in mixture model (null, small, medium, large effects)</p></td>
</tr>
<tr class="row-even"><td><p><strong>Alleles and Frequencies</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(A\)</span></p></td>
<td><p>Major allele</p></td>
<td><p>Reference allele (more common)</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(a\)</span></p></td>
<td><p>Minor allele</p></td>
<td><p>Alternative allele (less common)</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(f_A\)</span>, <span class="math notranslate nohighlight">\(f_a\)</span></p></td>
<td><p>Allele frequencies</p></td>
<td><p><span class="math notranslate nohighlight">\(f_A=0.7, f_a=0.3\)</span> for a variant</p></td>
</tr>
<tr class="row-even"><td><p><strong>Linkage and Relatedness</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(l_j\)</span></p></td>
<td><p>LD score of variant <span class="math notranslate nohighlight">\(j\)</span></p></td>
<td><p>Measure of local LD around variant <span class="math notranslate nohighlight">\(j\)</span></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathbf{G}\)</span></p></td>
<td><p>Genetic relationship matrix (GRM)</p></td>
<td><p><span class="math notranslate nohighlight">\(N \times N\)</span> matrix of genetic similarities</p></td>
</tr>
<tr class="row-odd"><td><p><strong>Traits and Phenotypes</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathbf{Y}\)</span></p></td>
<td><p>Trait vector (multivariate case)</p></td>
<td><p>Height and weight measurements</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(Y_i\)</span></p></td>
<td><p>Trait value for individual <span class="math notranslate nohighlight">\(i\)</span></p></td>
<td><p>Height = 175cm for individual <span class="math notranslate nohighlight">\(i\)</span></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(Y_{i,a}\)</span>, <span class="math notranslate nohighlight">\(Y_{i,b}\)</span></p></td>
<td><p>Trait values for individual <span class="math notranslate nohighlight">\(i\)</span> (multivariate)</p></td>
<td><p>Height = 175cm, weight = 70kg for individual <span class="math notranslate nohighlight">\(i\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><strong>Regression Parameters</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(b_0\)</span></p></td>
<td><p>Intercept in regression model</p></td>
<td><p>Mean height = 170cm when genotype = 0</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\beta\)</span></p></td>
<td><p>Effect size of variant on trait (scalar)</p></td>
<td><p>0.5cm increase in height per risk allele</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\boldsymbol{\beta}\)</span></p></td>
<td><p>Effect size vector (multivariate)</p></td>
<td><p>Effects on height and weight: <span class="math notranslate nohighlight">\(\begin{pmatrix} 0.5 \\ 0.3 \end{pmatrix}\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\hat{\beta}_k\)</span></p></td>
<td><p>Estimated effect from study <span class="math notranslate nohighlight">\(k\)</span></p></td>
<td><p>Effect estimate from European study</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mu_\beta\)</span></p></td>
<td><p>Mean of <span class="math notranslate nohighlight">\(\beta\)</span> distribution</p></td>
<td><p>Average effect across all variants</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(w_k\)</span></p></td>
<td><p>Weight for the <span class="math notranslate nohighlight">\(k\)</span>-th study under fixed effect meta-analysis</p></td>
<td><p><span class="math notranslate nohighlight">\(w_k = \frac{1}{\text{SE}_k^2}\)</span></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\tau^2\)</span></p></td>
<td><p>The between-study variance (heterogeneity) under random effect meta-analysis</p></td>
<td><p>How different the studies are from each other</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(w_k^*\)</span></p></td>
<td><p>Weight for the <span class="math notranslate nohighlight">\(k\)</span>-th study under random effect meta-analysis</p></td>
<td><p><span class="math notranslate nohighlight">\(w_k^* = \frac{1}{\text{SE}_k^2 + \tau^2}\)</span></p></td>
</tr>
<tr class="row-even"><td><p><strong>Variance and Covariance</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\sigma^2\)</span></p></td>
<td><p>Variance of trait</p></td>
<td><p>Residual variance in height (e.g., 100 cm²)</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\boldsymbol{\Sigma}\)</span></p></td>
<td><p>Known covariance matrix</p></td>
<td><p>Residual covariance between height and weight</p></td>
</tr>
<tr class="row-odd"><td><p><strong>Prior Parameters</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\beta_0\)</span></p></td>
<td><p>Prior mean (scalar)</p></td>
<td><p>Expected effect = 0 before seeing data</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\boldsymbol{\beta}_0\)</span></p></td>
<td><p>Prior mean vector</p></td>
<td><p>Expected effects = <span class="math notranslate nohighlight">\(\begin{pmatrix} 0 \\ 0 \end{pmatrix}\)</span> before data</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\sigma_0^2\)</span></p></td>
<td><p>Prior variance (scalar)</p></td>
<td><p>Prior uncertainty = 0.25 for effect size</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\boldsymbol{\Sigma}_0\)</span></p></td>
<td><p>Prior covariance matrix</p></td>
<td><p>Prior uncertainty about joint effects</p></td>
</tr>
<tr class="row-even"><td><p><strong>Posterior Parameters</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\beta_1\)</span></p></td>
<td><p>Posterior mean (scalar)</p></td>
<td><p>Updated effect = 0.3 after seeing data</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\boldsymbol{\beta}_1\)</span></p></td>
<td><p>Posterior mean vector</p></td>
<td><p>Updated effects = <span class="math notranslate nohighlight">\(\begin{pmatrix} 0.3 \\ 0.2 \end{pmatrix}\)</span> after data</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\sigma_1^2\)</span></p></td>
<td><p>Posterior variance (scalar)</p></td>
<td><p>Reduced uncertainty = 0.1 after data</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\boldsymbol{\Sigma}_1\)</span></p></td>
<td><p>Posterior covariance matrix</p></td>
<td><p>Reduced uncertainty about joint effects</p></td>
</tr>
<tr class="row-odd"><td><p><strong>Mixed Models</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathbf{g}\)</span></p></td>
<td><p>Random effect vector in linear mixed models</p></td>
<td><p>Individual-specific genetic effects</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\mathbf{Z}\)</span></p></td>
<td><p>Design matrix for random effects</p></td>
<td><p>Matrix linking observations to random effects</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathbf{u}\)</span></p></td>
<td><p>Random effect vector in linear mixed models</p></td>
<td><p>Polygenic effects for each individual</p></td>
</tr>
<tr class="row-odd"><td><p><strong>Covariates</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathbf{W}\)</span></p></td>
<td><p>Covariate matrix (confounder, collider, or mediator)</p></td>
<td><p>Age, sex, principal components</p></td>
</tr>
<tr class="row-odd"><td><p><strong>Meta-Analysis</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(Q\)</span></p></td>
<td><p>Cochran’s Q statistic</p></td>
<td><p>Test statistic for heterogeneity = 12.5</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(I^2\)</span></p></td>
<td><p><span class="math notranslate nohighlight">\(I^2\)</span> statistic for heterogeneity</p></td>
<td><p>Percentage of variation due to heterogeneity = 75%</p></td>
</tr>
<tr class="row-even"><td><p><strong>Model and Data</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(M\)</span></p></td>
<td><p>Model</p></td>
<td><p>Linear model vs. quadratic model</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(D\)</span></p></td>
<td><p>Observed data</p></td>
<td><p>Genotype and phenotype measurements</p></td>
</tr>
<tr class="row-odd"><td><p><strong>Likelihood and Testing</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\mathcal{L}\)</span></p></td>
<td><p>Likelihood function</p></td>
<td><p><span class="math notranslate nohighlight">\(\mathcal{L}(\beta\mid Y,X) = P(Y\mid X,\beta)\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\ell\)</span></p></td>
<td><p>Log-likelihood</p></td>
<td><p><span class="math notranslate nohighlight">\(\ell = \log(\mathcal{L})\)</span> for numerical stability</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\Lambda\)</span></p></td>
<td><p>Likelihood ratio test statistic</p></td>
<td><p><span class="math notranslate nohighlight">\(\Lambda = 2(\ell_1 - \ell_0)\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(p\)</span></p></td>
<td><p>P-value</p></td>
<td><p>Probability = 0.001 under null hypothesis</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(H_0\)</span></p></td>
<td><p>Null hypothesis</p></td>
<td><p>No genetic effect (<span class="math notranslate nohighlight">\(\beta = 0\)</span>)</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(H_a\)</span></p></td>
<td><p>Alternative hypothesis</p></td>
<td><p>Genetic effect exists (<span class="math notranslate nohighlight">\(\beta \neq 0\)</span>)</p></td>
</tr>
<tr class="row-even"><td><p><strong>Parameters and Spaces</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\theta\)</span></p></td>
<td><p>Generic parameter of statistical model</p></td>
<td><p>Could be <span class="math notranslate nohighlight">\(\beta\)</span>, <span class="math notranslate nohighlight">\(\sigma^2\)</span>, or other parameters</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\Theta\)</span></p></td>
<td><p>Parameter space of <span class="math notranslate nohighlight">\(\theta\)</span></p></td>
<td><p>Set of all possible values for <span class="math notranslate nohighlight">\(\theta\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><strong>Sufficient Statistics</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(T_1\)</span></p></td>
<td><p>Sufficient statistic (scalar)</p></td>
<td><p><span class="math notranslate nohighlight">\(T_1 = \sum_{i=1}^N X_i^2 = 1000\)</span></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\mathbf{T}_2\)</span></p></td>
<td><p>Sufficient statistic vector</p></td>
<td><p><span class="math notranslate nohighlight">\(\mathbf{T}_2 = \sum_{i=1}^N X_i \mathbf{Y}_i\)</span> for multivariate traits</p></td>
</tr>
<tr class="row-even"><td><p><strong>Mixture Models</strong></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\pi_c\)</span></p></td>
<td><p>Mixture weight for component <span class="math notranslate nohighlight">\(c\)</span></p></td>
<td><p>90% of variants have no effect (<span class="math notranslate nohighlight">\(\pi_1 = 0.9\)</span>)</p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(p_c(\beta)\)</span></p></td>
<td><p>Component density function</p></td>
<td><p>Normal distribution for component <span class="math notranslate nohighlight">\(c\)</span></p></td>
</tr>
</tbody>
</table>
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="README.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">statgen-prerequisites</p>
      </div>
    </a>
    <a class="right-next"
       href="popular_topics.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Popular Topics in Statistical Genetics</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Gao Wang
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>