
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Intuition &#8212; Statistical backgrounds for the Summer Program in Statistical Functional Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'unfinished_Bayesian_frequentist';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="README.html">
  
  
  
  
  
  
    <p class="title logo__title">Statistical backgrounds for the Summer Program in Statistical Functional Genomics</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Notations and References</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Notations.html"><strong>Notation Table</strong></a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Basic Quantities in Statistical Genetics</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="genotype_coding.html">Intuitional Description</a></li>





<li class="toctree-l1"><a class="reference internal" href="minor_allele_frequency.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="Hardy_Weinberg_equilibrium.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="linkage_disequilibrium_score.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="genetic_relationship_matrix.html">Intuitional Description</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Single Marker Linear Regression</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="ordinary_least_squares.html">Intuitional Description</a></li>





<li class="toctree-l1"><a class="reference internal" href="proportion_of_variance_explained.html">Intuitional Description</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Summary Statistics and Meta-Analysis</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="summary_statistics.html">Intuitional Description</a></li>






<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html">Intuitional Description</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Logistic Regression, Odds Ratio, and Mixed Models</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="odds_ratio.html">Intuitional Description</a></li>





<li class="toctree-l1"><a class="reference internal" href="linear_mixed_model.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="marginal_joint_effects.html">Intuitional Description</a></li>




</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Covariates in Genetic Studies</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="confounder.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="collider.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="mediator.html">Intuitional Description</a></li>





</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Likelihood Methods</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="likelihood.html">Intuitional Description</a></li>






<li class="toctree-l1"><a class="reference internal" href="maximum_likelihood_estimation.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio.html">Intuitional Description</a></li>







<li class="toctree-l1"><a class="reference internal" href="likelihood_ratio_test.html">Intuitional Description</a></li>







</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Bayesian Methods</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Bayesian_normal_mean_model.html">Intuitional Description</a></li>





<li class="toctree-l1"><a class="reference internal" href="Bayesian_multivariate_normal_mean_model.html">Intuitional Description</a></li>





<li class="toctree-l1"><a class="reference internal" href="p_value_versus_Bayes_factor.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="Bayes_factor.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_comparison.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="Bayesian_mixture_model.html">Intuitional Description</a></li>




<li class="toctree-l1"><a class="reference internal" href="Bayesian_model_averaging.html">Intuitional Description</a></li>




</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/unfinished_Bayesian_frequentist.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Intuition</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Intuition</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#notations">Notations</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bayes-factor-bf">Bayes Factor (BF)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#advantages-of-bayesian-analysis-over-frequentist-methods">Advantages of Bayesian Analysis Over Frequentist Methods</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#example">Example</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#frequentist">Frequentist</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-lrt-to-reject-the-hypothesis-p-value">From LRT to reject the hypothesis (<em>p-value</em>)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#mle">MLE</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bayesian">Bayesian</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#when-we-have-the-prior-of-the-model-and-it-is-informative">When we have the prior of the model and it is informative …</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#when-the-model-is-not-fully-specific">When the model is not fully specific</a></li>
</ul>
</li>
</ul>
</li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="intuition">
<h1>Intuition<a class="headerlink" href="#intuition" title="Link to this heading">#</a></h1>
<p>Bayesian vs frequentist inference</p>
<p><img alt="figure" src="_images/8.svg" /></p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="notations">
<h1>Notations<a class="headerlink" href="#notations" title="Link to this heading">#</a></h1>
<section id="bayes-factor-bf">
<h2>Bayes Factor (BF)<a class="headerlink" href="#bayes-factor-bf" title="Link to this heading">#</a></h2>
<p>The <strong>Bayes Factor (BF)</strong> (sometimes denoted as <span class="math notranslate nohighlight">\(K\)</span>) is defined as the ratio of the marginal likelihoods of two models, taking into account the prior distributions over the models’ parameters. It is calculated as:</p>
<div class="math notranslate nohighlight">
\[
BF(M_2, M_1) := \frac{Pr(D|M_2)}{Pr(D|M_1)} = \frac{\int Pr(D|\theta_2, M_2) Pr(\theta_2|M_2) d\theta_2}{\int Pr(D|\theta_1, M_1) Pr(\theta_1|M_1) d\theta_1} = \frac{\frac{Pr(M_2|D)Pr(D)}{Pr(M_2)} }{\frac{Pr(M_1|D)Pr(D)}{Pr(M_1)}} = \frac{Pr(M_2|D)}{Pr(M_1|D)} \times \frac{Pr(M_1)}{Pr(M_2)}
\]</div>
<p>The Bayes Factor compares the marginal likelihoods of the models after considering the uncertainty in the parameters, <strong>weighted by the prior beliefs of the model</strong>.</p>
<p><strong>Comparison between LR and BF</strong></p>
<p>When informative priors are <strong>absent</strong> or when <strong>non-informative priors</strong> (i.e., flat priors) are used, the prior distributions do not influence the marginal likelihoods. In such a case, the integration over the parameters becomes irrelevant, and the marginal likelihood simplifies to just the likelihood itself.</p>
<p>Therefore, the formula for BF becomes identical to the formula for LR (<span class="math notranslate nohighlight">\(Pr(M_1) = Pr(M_2)\)</span>):</p>
<div class="math notranslate nohighlight">
\[
BF(M_2, M_1) = \frac{p(D|M_2)}{p(D|M_1)} = \frac{Pr(M_2|D)}{Pr(M_1|D)} \times \frac{Pr(M_1)}{Pr(M_2)}= \frac{L(M_2)}{L(M_1)} = LR(M_2, M_1)
\]</div>
<p>For fully specified models (In statistical modeling, a fully specified model refers to a model where all parameters and components are clearly defined), the <strong>likelihood ratio</strong> is equivalent to the <strong>Bayes Factor</strong>. When <strong>non-informative priors</strong> are used, LR and BF are numerically identical because the priors cancel out. However, when <strong>informative priors</strong> are included, BF will differ from LR because it incorporates prior beliefs about the parameters, whereas LR only considers the likelihood based on the data.</p>
<ul class="simple">
<li><p><strong>Likelihood Ratio (LR)</strong> compares two models using the <strong>likelihood function</strong> of the models, ignoring prior distributions. It is a <strong>frequentist</strong> concept.</p></li>
<li><p><strong>Bayes Factor (BF)</strong>, on the other hand, compares two models by integrating over the parameters of the models, taking into account their <strong>prior distributions</strong>. It is a <strong>Bayesian</strong> concept.</p></li>
</ul>
</section>
<section id="advantages-of-bayesian-analysis-over-frequentist-methods">
<h2>Advantages of Bayesian Analysis Over Frequentist Methods<a class="headerlink" href="#advantages-of-bayesian-analysis-over-frequentist-methods" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><strong>Incorporation of Prior Knowledge</strong></p>
<ul>
<li><p><strong>Frequentist Limitation:</strong> Frequentist methods rely solely on observed data and do not incorporate prior information, making estimates unstable in small sample sizes.</p></li>
<li><p><strong>Bayesian Advantage:</strong> Bayesian analysis integrates prior knowledge through a prior distribution, improving inference when data are limited or prior information is available.</p></li>
</ul>
</li>
<li><p><strong>Probabilistic Interpretation</strong></p>
<ul>
<li><p><strong>Frequentist Limitation:</strong> Confidence intervals do not provide a direct probability interpretation; they describe long-run frequency properties rather than certainty about parameters.</p></li>
<li><p><strong>Bayesian Advantage:</strong> Bayesian credible intervals directly represent the probability that a parameter lies within a given range, making interpretation more intuitive.</p></li>
</ul>
</li>
<li><p><strong>Flexibility in Complex Models</strong></p>
<ul>
<li><p><strong>Frequentist Limitation:</strong> Frequentist methods often require separate estimation steps or computational approximations for hierarchical models, missing data, and latent variables.</p></li>
<li><p><strong>Bayesian Advantage:</strong> Bayesian methods seamlessly integrate over uncertainties, allowing for more flexible modeling of hierarchical structures and missing data.</p></li>
</ul>
</li>
<li><p><strong>Direct Model Comparison</strong></p>
<ul>
<li><p><strong>Frequentist Limitation:</strong> Frequentist model comparison relies on likelihood ratio tests (LRT) or information criteria (AIC, BIC), which do not naturally integrate over parameter uncertainty.</p></li>
<li><p><strong>Bayesian Advantage:</strong> Bayesian model comparison uses Bayes factors, which account for both model fit and complexity by integrating over parameter uncertainty, avoiding overfitting.</p></li>
</ul>
</li>
</ul>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="example">
<h1>Example<a class="headerlink" href="#example" title="Link to this heading">#</a></h1>
<p>In example 6.1 and 6.2, we compare two model <span class="math notranslate nohighlight">\(M_1\)</span> (<span class="math notranslate nohighlight">\(\theta=0\)</span>) and <span class="math notranslate nohighlight">\(M_2\)</span> (<span class="math notranslate nohighlight">\(\theta=2\)</span>) and calculate the LRT.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="o">=</span><span class="nf">ls</span><span class="p">())</span>
<span class="nf">set.seed</span><span class="p">(</span><span class="m">2</span><span class="p">)</span>

<span class="c1"># Simulate true mean and effect size</span>
<span class="n">baseline</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">170</span><span class="w">  </span><span class="c1"># Population mean of the trait (e.g., height in cm) when the genetic variant has no effect (Model 1)</span>
<span class="n">theta_true</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span><span class="w">  </span><span class="c1"># True effect size of the genetic variant. This represents the change in height (in cm) associated with each additional minor allele (Model 2)</span>
<span class="n">sd_y</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Standard deviation of the trait (e.g., variability in height measurement within the population)</span>

<span class="c1"># Simulate genotypes for three individuals</span>
<span class="n">genotype</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>

<span class="c1"># Simulate height values for three individuals based on genotypes</span>
<span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">genotype</span><span class="p">)</span>
<span class="n">height_values</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rnorm</span><span class="p">(</span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">baseline</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theta_true</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd_y</span><span class="p">)</span>

<span class="c1"># Display the simulated data</span>
<span class="nf">data.frame</span><span class="p">(</span><span class="n">genotype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">height_values</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 3 × 2</caption>
<thead>
	<tr><th scope=col>genotype</th><th scope=col>height</th></tr>
	<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>1</td><td>171.1031</td></tr>
	<tr><td>2</td><td>174.1848</td></tr>
	<tr><td>0</td><td>171.5878</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Define the likelihood function for a normal distribution</span>
<span class="n">likelihood</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">theta</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">prod</span><span class="p">(</span><span class="nf">dnorm</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">theta</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd</span><span class="p">))</span>
<span class="p">}</span>

<span class="c1"># Likelihood for Model 1: No genetic effect</span>
<span class="n">likelihood_1_multi_samples</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">likelihood</span><span class="p">(</span><span class="n">theta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">baseline</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd_y</span><span class="p">,</span><span class="w"> </span><span class="n">height_values</span><span class="p">)</span>

<span class="c1"># Likelihood for Model 2: Genetic effect</span>
<span class="n">likelihood_2_multi_samples</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">likelihood</span><span class="p">(</span><span class="n">theta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">baseline</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theta_true</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd_y</span><span class="p">,</span><span class="w"> </span><span class="n">height_values</span><span class="p">)</span>

<span class="c1"># Print the likelihoods for both models</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Likelihood for Model 1:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">likelihood_1_multi_samples</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Likelihood for Model 2:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">likelihood_2_multi_samples</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="c1"># Compute the Likelihood Ratio Test (LRT) statistic</span>
<span class="n">lrt_statistic</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">-2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="nf">log</span><span class="p">(</span><span class="n">likelihood_1_multi_samples</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">log</span><span class="p">(</span><span class="n">likelihood_2_multi_samples</span><span class="p">))</span>

<span class="c1"># Print the LRT statistic</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Likelihood Ratio Test Statistic (LRT):&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lrt_statistic</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Likelihood for Model 1: 1.542167e-06 
Likelihood for Model 2: 0.01183428 
Likelihood Ratio Test Statistic (LRT): 17.89114 
</pre></div>
</div>
</div>
</div>
<section id="frequentist">
<h2>Frequentist<a class="headerlink" href="#frequentist" title="Link to this heading">#</a></h2>
<section id="from-lrt-to-reject-the-hypothesis-p-value">
<h3>From LRT to reject the hypothesis (<em>p-value</em>)<a class="headerlink" href="#from-lrt-to-reject-the-hypothesis-p-value" title="Link to this heading">#</a></h3>
<p>Generally what a frequentist do next is to <strong>perform the likelihood ratio statistics</strong>. The likelihood ratio statistic can be compared to the <strong>chi-square distribution with 1 degree of freedom</strong>. If the test statistic is large enough (i.e., exceeds the critical value from the chi-square distribution at a chosen significance level, say 0.05), then we reject <span class="math notranslate nohighlight">\(H_1\)</span> in favor of <span class="math notranslate nohighlight">\(H_2\)</span>, indicating that the genetic variant has a significant effect on height.</p>
<blockquote>
<div><p>Then one would like to address the question: <strong>what value of the LR should we treat as “convincing” evidence for one model vs another?</strong> For example, in statistics there are many situations where we want to compare models that are not equally plausible. Suppose that model <span class="math notranslate nohighlight">\(M_2\)</span> is much less plausible than <span class="math notranslate nohighlight">\(M_1\)</span>. Then we must surely demand stronger evidence from the data (larger LR) to be “convinced” that it arose from model <span class="math notranslate nohighlight">\(M_2\)</span> rather than <span class="math notranslate nohighlight">\(M_1\)</span>, than in contexts where <span class="math notranslate nohighlight">\(M_2\)</span> and <span class="math notranslate nohighlight">\(M_1\)</span> are equally plausible.</p>
<p>It is crucial to recognize that the answer to this question has to be context dependent. In particular, the extent to which we should be “convinced” by a particular LR value has to depend on the relative plausibility of the models we are comparing.</p>
</div></blockquote>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Compare with chi-square distribution with 1 degree of freedom</span>
<span class="n">p_value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pchisq</span><span class="p">(</span><span class="n">lrt_statistic</span><span class="p">,</span><span class="w"> </span><span class="n">df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">lower.tail</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="c1"># Print result</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;LRT statistic:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lrt_statistic</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;p-value:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">p_value</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>

<span class="c1"># Interpretation</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">p_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Reject H0: The genetic variant has a significant effect on height.\n&quot;</span><span class="p">)</span>
<span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Fail to reject H0: The genetic variant has no significant effect on height.\n&quot;</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>LRT statistic: 17.89114 
p-value: 2.339081e-05 
Reject H0: The genetic variant has a significant effect on height.
</pre></div>
</div>
</div>
</div>
</section>
<section id="mle">
<h3>MLE<a class="headerlink" href="#mle" title="Link to this heading">#</a></h3>
<p>In statistics, maximum likelihood estimation (MLE) is a method of estimating the parameters of an assumed probability distribution, given some observed data. This is achieved by maximizing a likelihood function so that, under the assumed statistical model, the observed data is most probable. The point in the parameter space that maximizes the likelihood function is called the <strong>maximum likelihood estimate</strong>.</p>
<p>Now let’s assume that <span class="math notranslate nohighlight">\(\theta\)</span> actually follows some distribution and for each value of <span class="math notranslate nohighlight">\(\theta\)</span> we can calculate the corresponding likelihood <span class="math notranslate nohighlight">\(L(\theta)\)</span>. Then what is the value when <span class="math notranslate nohighlight">\(L(\theta)\)</span> reaches the maximum? The value of <span class="math notranslate nohighlight">\(\theta\)</span> that maximizes the likelihood function is referred to as the <strong>maximum likelihood estimate</strong>, and usually denoted <span class="math notranslate nohighlight">\(\hat{\theta}\)</span>, i.e.,</p>
<div class="math notranslate nohighlight">
\[
\hat{\theta}:= \text{argmax} L(\theta)
\]</div>
<p>Since
$<span class="math notranslate nohighlight">\(
LRT(M_2, M_1) := - 2 \log \left( \frac{L(M_1)}{L(M_2)} \right) = - 2 \left[ \log L(M_1) - \log L(M_2) \right]
\)</span>$</p>
<p>it is equivalent to finding the minimum value of the LRT.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Define the negative log-likelihood function for optimization</span>
<span class="n">neg_log_likelihood</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">theta</span><span class="p">,</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="p">,</span><span class="w"> </span><span class="n">baseline</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">n</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">genotype</span><span class="p">)</span>
<span class="w">      </span><span class="n">log_likelihood_val</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">dnorm</span><span class="p">(</span><span class="n">height</span><span class="p">,</span><span class="w"> </span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">baseline</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theta</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd</span><span class="p">,</span><span class="w"> </span><span class="n">log</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
<span class="w">  </span><span class="kr">return</span><span class="p">(</span><span class="m">-2</span><span class="o">*</span><span class="n">log_likelihood_val</span><span class="p">)</span>
<span class="p">}</span>

<span class="c1"># Estimate theta using MLE (optimizing the negative log-likelihood)</span>
<span class="n">mle_result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">optimize</span><span class="p">(</span><span class="n">neg_log_likelihood</span><span class="p">,</span><span class="w"> </span><span class="n">interval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-5</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="p">),</span><span class="w"> </span><span class="n">genotype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">height_values</span><span class="p">,</span><span class="w"> </span><span class="n">baseline</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">baseline</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd_y</span><span class="p">)</span>

<span class="c1"># Extract the estimated theta</span>
<span class="n">theta_mle</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mle_result</span><span class="o">$</span><span class="n">minimum</span>

<span class="nf">print</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;The MLE of theta is &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">theta_mle</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;The MLE of theta is 1.8945567645337&quot;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Plot the log-likelihood function</span>
<span class="n">theta_vals</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">seq</span><span class="p">(</span><span class="m">-5</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">length.out</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="n">log_likelihood_vals</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="n">theta_vals</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">theta</span><span class="p">)</span><span class="w"> </span><span class="nf">neg_log_likelihood</span><span class="p">(</span><span class="n">theta</span><span class="p">,</span><span class="w"> </span><span class="n">genotype</span><span class="p">,</span><span class="w"> </span><span class="n">height_values</span><span class="p">,</span><span class="w"> </span><span class="n">baseline</span><span class="p">,</span><span class="w"> </span><span class="n">sd_y</span><span class="p">))</span>

<span class="c1"># Plot the log-likelihood function</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">theta_vals</span><span class="p">,</span><span class="w"> </span><span class="n">log_likelihood_vals</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;l&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span>
<span class="w">     </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Theta (Genetic Effect)&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Log-Likelihood&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">     </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Log-Likelihood Function and MLE&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">cex.lab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.5</span><span class="p">,</span><span class="w"> </span><span class="n">cex.main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.5</span><span class="p">)</span>

<span class="c1"># Add a vertical line at the MLE</span>
<span class="nf">abline</span><span class="p">(</span><span class="n">v</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">theta_mle</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>

<span class="c1"># Add text to label the MLE</span>
<span class="nf">text</span><span class="p">(</span><span class="n">theta_mle</span><span class="p">,</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">log_likelihood_vals</span><span class="p">),</span><span class="w"> </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;MLE = &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">theta_mle</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)),</span><span class="w"> </span><span class="n">pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.2</span><span class="p">)</span>

<span class="c1"># Show the plot</span>
<span class="nf">grid</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="_images/d1295a70cf0edb2b88830d748b281d1b9f75bbc164220f55c3f63c12845d5fb7.png"><img alt="_images/d1295a70cf0edb2b88830d748b281d1b9f75bbc164220f55c3f63c12845d5fb7.png" src="_images/d1295a70cf0edb2b88830d748b281d1b9f75bbc164220f55c3f63c12845d5fb7.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
</section>
<section id="bayesian">
<h2>Bayesian<a class="headerlink" href="#bayesian" title="Link to this heading">#</a></h2>
<p>Now let’s see how a Bayesian statistician would handle this question.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Likelihood for Model 1:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">likelihood_1_multi_samples</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Likelihood for Model 2:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">likelihood_2_multi_samples</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Likelihood for Model 1: 1.542167e-06 
Likelihood for Model 2: 0.01183428 
</pre></div>
</div>
</div>
</div>
<section id="when-we-have-the-prior-of-the-model-and-it-is-informative">
<h3>When we have the prior of the model and it is informative …<a class="headerlink" href="#when-we-have-the-prior-of-the-model-and-it-is-informative" title="Link to this heading">#</a></h3>
<p>If we don’t know anything about the probability of <span class="math notranslate nohighlight">\(M_1\)</span> and <span class="math notranslate nohighlight">\(M_2\)</span>, then it is generally assumed that they are equally possible:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Compute the Bayes Factor</span>
<span class="n">bayes_factor</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">likelihood_2_multi_samples</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">likelihood_1_multi_samples</span>

<span class="c1"># Print the Bayes Factor</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Bayes Factor (without informative priors):&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">bayes_factor</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Bayes Factor (without informative priors): 7673.803 
</pre></div>
</div>
</div>
</div>
<p>In this case the result is the same as the LRT that we calcualte above.</p>
<p>However when we are aware of a fact that <strong>this genetic variant actually comes from a gene that’s been proved to be directly associated with height</strong>, then we believe that the probability of <span class="math notranslate nohighlight">\(M_1\)</span> is lower than the probability of <span class="math notranslate nohighlight">\(M_2\)</span>. Using Bayesian analysis allows us to incorporate this information into the inference:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Define prior probabilities for Model 1 and Model 2</span>
<span class="n">prior_M1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0.3</span><span class="w">  </span><span class="c1"># Prior probability for Model 1</span>
<span class="n">prior_M2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0.7</span><span class="w">  </span><span class="c1"># Prior probability for Model 2</span>

<span class="c1"># Compute the adjusted Bayes Factor</span>
<span class="n">bayes_factor</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">prior_M2</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">prior_M1</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="n">likelihood_2_multi_samples</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">likelihood_1_multi_samples</span><span class="p">)</span>

<span class="c1"># Print the adjusted Bayes Factor</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Bayes Factor (with priors):&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">bayes_factor</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Bayes Factor (with priors): 17905.54 
</pre></div>
</div>
</div>
</div>
</section>
<section id="when-the-model-is-not-fully-specific">
<h3>When the model is not fully specific<a class="headerlink" href="#when-the-model-is-not-fully-specific" title="Link to this heading">#</a></h3>
<p>Unlike a likelihood-ratio test, this Bayesian model comparison does not depend on any single set of parameters, as it integrates over all parameters in each model (with respect to the respective priors). An advantage of the use of Bayes factors is that it automatically, and quite naturally, includes a penalty for including too much model structure</p>
<p>For example, if we want to compare <span class="math notranslate nohighlight">\(M_1\)</span> and <span class="math notranslate nohighlight">\(M^{'}_2\)</span>,</p>
<ul class="simple">
<li><p><strong>Model 1 (<span class="math notranslate nohighlight">\(M_1\)</span>)</strong>: The genetic variant has <strong>no effect</strong> on height.</p></li>
<li><p><strong>Model 2’ (<span class="math notranslate nohighlight">\(M^{'}_2\)</span>)</strong>: The genetic variant has a <strong>significant effect</strong> between 1 and 3 on height.</p></li>
</ul>
<p>In frequentist analysis, we typically perform a Likelihood Ratio Test (LRT) to compare nested models. However, for this comparison between <span class="math notranslate nohighlight">\(M_1\)</span> and <span class="math notranslate nohighlight">\(M^{'}_2\)</span>, the models are not nested. Model 2’ introduces a parameter space (effect size range of 1 to 3) that is not a simple extension of Model 1. Because of this, LRT is not appropriate for comparing these two models, as it requires nested models where one model is a special case of the other (i.e., <span class="math notranslate nohighlight">\(M_1\)</span> must be a simpler version of <span class="math notranslate nohighlight">\(M^{'}_2\)</span>). Therefore, in this case, the LRT will not work.</p>
<p>On the other hand, Bayesian analysis can still be applied because it doesn’t require nested models. In Bayesian model comparison, we calculate the Bayes Factor (BF) to assess the relative support for the two models. Bayes factors naturally handle model comparison even for non-nested models by integrating over all parameters and incorporating the prior knowledge.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create a vectorized function for integration</span>
<span class="n">likelihood_function</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">theta_value</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="c1"># Calculate the expected value for each observation based on genotype</span>
<span class="w">    </span><span class="n">means</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">baseline</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">theta_value</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">genotype</span>
<span class="w">    </span><span class="c1"># Calculate likelihood for this theta</span>
<span class="w">    </span><span class="n">result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">likelihood</span><span class="p">(</span><span class="n">theta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">means</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sd_y</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">height_values</span><span class="p">)</span>
<span class="w">  </span><span class="kr">return</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
<span class="p">}</span>
<span class="n">integrated_likelihood_M2_prime</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">integrate</span><span class="p">(</span><span class="nf">Vectorize</span><span class="p">(</span><span class="n">likelihood_function</span><span class="p">),</span><span class="w"> </span><span class="n">lower</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">upper</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span>
<span class="n">integrated_likelihood_M2_prime</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Integrated likelihood for Model 2&#39;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">integrated_likelihood_M2_prime</span><span class="o">$</span><span class="n">value</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>0.01323835 with absolute error &lt; 2.1e-12
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Integrated likelihood for Model 2&#39;: 0.0132383537758094&quot;
</pre></div>
</div>
</div>
</div>
<p>Then the bayes factor between <span class="math notranslate nohighlight">\(M_1\)</span> and <span class="math notranslate nohighlight">\(M^{'}_2\)</span> is:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Compute the adjusted Bayes Factor</span>
<span class="n">bayes_factor</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">integrated_likelihood_M2_prime</span><span class="o">$</span><span class="n">value</span><span class="o">/</span><span class="n">likelihood_1_multi_samples</span>

<span class="c1"># Print the adjusted Bayes Factor</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Bayes Factor (M1 versus M2&#39;):&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">bayes_factor</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Bayes Factor (M1 versus M2&#39;): 8584.255 
</pre></div>
</div>
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Intuition</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#notations">Notations</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bayes-factor-bf">Bayes Factor (BF)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#advantages-of-bayesian-analysis-over-frequentist-methods">Advantages of Bayesian Analysis Over Frequentist Methods</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#example">Example</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#frequentist">Frequentist</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#from-lrt-to-reject-the-hypothesis-p-value">From LRT to reject the hypothesis (<em>p-value</em>)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#mle">MLE</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bayesian">Bayesian</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#when-we-have-the-prior-of-the-model-and-it-is-informative">When we have the prior of the model and it is informative …</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#when-the-model-is-not-fully-specific">When the model is not fully specific</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Gao Wang
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>